{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "runtime_attributes": {
        "runtime_version": "2025.07"
      },
      "authorship_tag": "ABX9TyM7yVNrZTVJ25Bmpx6D+XON",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/keulreobeu/deep_learning_for_the_life_sciences/blob/main/%EC%9C%A0%EC%A0%84%ED%95%99%EA%B3%BC_%EB%94%A5%EB%9F%AC%EB%8B%9D.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "a2CMUXAvnot3"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -U pip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mrQxjrMk7Qd7",
        "outputId": "d54ef189-d784-4aaa-88a4-b24f167143f4"
      },
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: pip in /usr/local/lib/python3.11/dist-packages (24.1.2)\n",
            "Collecting pip\n",
            "  Downloading pip-25.2-py3-none-any.whl.metadata (4.7 kB)\n",
            "Downloading pip-25.2-py3-none-any.whl (1.8 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.8/1.8 MB\u001b[0m \u001b[31m13.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: pip\n",
            "  Attempting uninstall: pip\n",
            "    Found existing installation: pip 24.1.2\n",
            "    Uninstalling pip-24.1.2:\n",
            "      Successfully uninstalled pip-24.1.2\n",
            "Successfully installed pip-25.2\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install --pre deepchem"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Uji16CeWoeNb",
        "outputId": "56b993be-63cc-4495-d9ab-38f7852b6f23"
      },
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: deepchem in /usr/local/lib/python3.11/dist-packages (2.8.1.dev20250829211407)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from deepchem) (1.5.1)\n",
            "Requirement already satisfied: numpy<2 in /usr/local/lib/python3.11/dist-packages (from deepchem) (1.26.4)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from deepchem) (2.2.2)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.11/dist-packages (from deepchem) (1.6.1)\n",
            "Requirement already satisfied: sympy in /usr/local/lib/python3.11/dist-packages (from deepchem) (1.13.1)\n",
            "Requirement already satisfied: scipy>=1.10.1 in /usr/local/lib/python3.11/dist-packages (from deepchem) (1.15.3)\n",
            "Requirement already satisfied: rdkit in /usr/local/lib/python3.11/dist-packages (from deepchem) (2025.3.6)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.11/dist-packages (from pandas->deepchem) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->deepchem) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas->deepchem) (2025.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.8.2->pandas->deepchem) (1.17.0)\n",
            "Requirement already satisfied: Pillow in /usr/local/lib/python3.11/dist-packages (from rdkit->deepchem) (11.2.1)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.11/dist-packages (from scikit-learn->deepchem) (3.6.0)\n",
            "Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.11/dist-packages (from sympy->deepchem) (1.3.0)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "런타임 재시작 후 다시실행\n"
      ],
      "metadata": {
        "id": "I4rDyzi4_vFf"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install \"keras<3\" \"tensorflow<2.16\""
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FcTvyGu094kD",
        "outputId": "b40b7a2c-097f-4d62-fedd-39ca1707f02a"
      },
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: keras<3 in /usr/local/lib/python3.11/dist-packages (2.15.0)\n",
            "Requirement already satisfied: tensorflow<2.16 in /usr/local/lib/python3.11/dist-packages (2.15.1)\n",
            "Requirement already satisfied: absl-py>=1.0.0 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (1.4.0)\n",
            "Requirement already satisfied: astunparse>=1.6.0 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (1.6.3)\n",
            "Requirement already satisfied: flatbuffers>=23.5.26 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (25.2.10)\n",
            "Requirement already satisfied: gast!=0.5.0,!=0.5.1,!=0.5.2,>=0.2.1 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (0.6.0)\n",
            "Requirement already satisfied: google-pasta>=0.1.1 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (0.2.0)\n",
            "Requirement already satisfied: h5py>=2.9.0 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (3.14.0)\n",
            "Requirement already satisfied: libclang>=13.0.0 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (18.1.1)\n",
            "Requirement already satisfied: ml-dtypes~=0.3.1 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (0.3.2)\n",
            "Requirement already satisfied: numpy<2.0.0,>=1.23.5 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (1.26.4)\n",
            "Requirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (3.4.0)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (24.2)\n",
            "Requirement already satisfied: protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (4.25.8)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (75.2.0)\n",
            "Requirement already satisfied: six>=1.12.0 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (1.17.0)\n",
            "Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (3.1.0)\n",
            "Requirement already satisfied: typing-extensions>=3.6.6 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (4.14.1)\n",
            "Requirement already satisfied: wrapt<1.15,>=1.11.0 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (1.14.2)\n",
            "Requirement already satisfied: tensorflow-io-gcs-filesystem>=0.23.1 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (0.37.1)\n",
            "Requirement already satisfied: grpcio<2.0,>=1.24.3 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (1.73.1)\n",
            "Requirement already satisfied: tensorboard<2.16,>=2.15 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (2.15.2)\n",
            "Requirement already satisfied: tensorflow-estimator<2.16,>=2.15.0 in /usr/local/lib/python3.11/dist-packages (from tensorflow<2.16) (2.15.0)\n",
            "Requirement already satisfied: google-auth<3,>=1.6.3 in /usr/local/lib/python3.11/dist-packages (from tensorboard<2.16,>=2.15->tensorflow<2.16) (2.38.0)\n",
            "Requirement already satisfied: google-auth-oauthlib<2,>=0.5 in /usr/local/lib/python3.11/dist-packages (from tensorboard<2.16,>=2.15->tensorflow<2.16) (1.2.2)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.11/dist-packages (from tensorboard<2.16,>=2.15->tensorflow<2.16) (3.8.2)\n",
            "Requirement already satisfied: requests<3,>=2.21.0 in /usr/local/lib/python3.11/dist-packages (from tensorboard<2.16,>=2.15->tensorflow<2.16) (2.32.3)\n",
            "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /usr/local/lib/python3.11/dist-packages (from tensorboard<2.16,>=2.15->tensorflow<2.16) (0.7.2)\n",
            "Requirement already satisfied: werkzeug>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from tensorboard<2.16,>=2.15->tensorflow<2.16) (3.1.3)\n",
            "Requirement already satisfied: cachetools<6.0,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from google-auth<3,>=1.6.3->tensorboard<2.16,>=2.15->tensorflow<2.16) (5.5.2)\n",
            "Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.11/dist-packages (from google-auth<3,>=1.6.3->tensorboard<2.16,>=2.15->tensorflow<2.16) (0.4.2)\n",
            "Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.11/dist-packages (from google-auth<3,>=1.6.3->tensorboard<2.16,>=2.15->tensorflow<2.16) (4.9.1)\n",
            "Requirement already satisfied: requests-oauthlib>=0.7.0 in /usr/local/lib/python3.11/dist-packages (from google-auth-oauthlib<2,>=0.5->tensorboard<2.16,>=2.15->tensorflow<2.16) (2.0.0)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests<3,>=2.21.0->tensorboard<2.16,>=2.15->tensorflow<2.16) (3.4.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests<3,>=2.21.0->tensorboard<2.16,>=2.15->tensorflow<2.16) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests<3,>=2.21.0->tensorboard<2.16,>=2.15->tensorflow<2.16) (2.4.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests<3,>=2.21.0->tensorboard<2.16,>=2.15->tensorflow<2.16) (2025.7.14)\n",
            "Requirement already satisfied: pyasn1>=0.1.3 in /usr/local/lib/python3.11/dist-packages (from rsa<5,>=3.1.4->google-auth<3,>=1.6.3->tensorboard<2.16,>=2.15->tensorflow<2.16) (0.6.1)\n",
            "Requirement already satisfied: wheel<1.0,>=0.23.0 in /usr/local/lib/python3.11/dist-packages (from astunparse>=1.6.0->tensorflow<2.16) (0.45.1)\n",
            "Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.11/dist-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<2,>=0.5->tensorboard<2.16,>=2.15->tensorflow<2.16) (3.3.1)\n",
            "Requirement already satisfied: MarkupSafe>=2.1.1 in /usr/local/lib/python3.11/dist-packages (from werkzeug>=1.0.1->tensorboard<2.16,>=2.15->tensorflow<2.16) (3.0.2)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "런타임 재시작 후 다시실행"
      ],
      "metadata": {
        "id": "uEcgnCMnEivf"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install rdkit"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "564VOYaH7TMq",
        "outputId": "95815122-3d6a-4fae-ae9b-d654d6281778"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: rdkit in /usr/local/lib/python3.11/dist-packages (2025.3.6)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from rdkit) (1.26.4)\n",
            "Requirement already satisfied: Pillow in /usr/local/lib/python3.11/dist-packages (from rdkit) (11.2.1)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip list | egrep 'deepchem|rdkit|tensorflow|keras|numpy|pandas|scikit-learn|matplotlib'\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "T-DFVOA5aLuP",
        "outputId": "408c5f10-264d-40d0-facc-357eaf042aeb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "deepchem                              2.8.1.dev20250829211407\n",
            "geopandas                             1.0.1\n",
            "keras                                 2.15.0\n",
            "keras-hub                             0.18.1\n",
            "keras-nlp                             0.18.1\n",
            "matplotlib                            3.10.0\n",
            "matplotlib-inline                     0.1.7\n",
            "matplotlib-venn                       1.1.2\n",
            "numpy                                 1.26.4\n",
            "pandas                                2.2.2\n",
            "pandas-datareader                     0.10.0\n",
            "pandas-gbq                            0.29.2\n",
            "pandas-stubs                          2.2.2.240909\n",
            "rdkit                                 2025.3.6\n",
            "scikit-learn                          1.6.1\n",
            "sklearn-pandas                        2.2.0\n",
            "tensorflow                            2.15.1\n",
            "tensorflow-datasets                   4.9.9\n",
            "tensorflow_decision_forests           1.11.0\n",
            "tensorflow-estimator                  2.15.0\n",
            "tensorflow-hub                        0.16.1\n",
            "tensorflow-io-gcs-filesystem          0.37.1\n",
            "tensorflow-metadata                   1.17.2\n",
            "tensorflow-probability                0.25.0\n",
            "tensorflow-text                       2.18.1\n",
            "tf_keras                              2.18.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import deepchem as dc\n",
        "from rdkit import Chem\n",
        "import numpy as np\n",
        "import pandas as pd"
      ],
      "metadata": {
        "id": "LAusRNo1EROm",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "7ea28127-31bd-4199-9104-0f7519eaecd1"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "WARNING:deepchem.feat.molecule_featurizers.rdkit_descriptors:No normalization for SPS. Feature removed!\n",
            "WARNING:deepchem.feat.molecule_featurizers.rdkit_descriptors:No normalization for AvgIpc. Feature removed!\n",
            "WARNING:deepchem.feat.molecule_featurizers.rdkit_descriptors:No normalization for NumAmideBonds. Feature removed!\n",
            "WARNING:deepchem.feat.molecule_featurizers.rdkit_descriptors:No normalization for NumAtomStereoCenters. Feature removed!\n",
            "WARNING:deepchem.feat.molecule_featurizers.rdkit_descriptors:No normalization for NumBridgeheadAtoms. Feature removed!\n",
            "WARNING:deepchem.feat.molecule_featurizers.rdkit_descriptors:No normalization for NumHeterocycles. Feature removed!\n",
            "WARNING:deepchem.feat.molecule_featurizers.rdkit_descriptors:No normalization for NumSpiroAtoms. Feature removed!\n",
            "WARNING:deepchem.feat.molecule_featurizers.rdkit_descriptors:No normalization for NumUnspecifiedAtomStereoCenters. Feature removed!\n",
            "WARNING:deepchem.feat.molecule_featurizers.rdkit_descriptors:No normalization for Phi. Feature removed!\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: \u001b[33mWARNING\u001b[0m W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.11/dist-packages/tensorflow/python/util/deprecation.py:588: calling function (from tensorflow.python.eager.polymorphic_function.polymorphic_function) with experimental_relax_shapes is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "experimental_relax_shapes is deprecated, use reduce_retracing instead\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: \u001b[33mWARNING\u001b[0m W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.\n",
            "WARNING:deepchem.models.torch_models:Skipped loading modules with pytorch-geometric dependency, missing a dependency. No module named 'torch_geometric'\n",
            "WARNING:deepchem.models:Skipped loading modules with pytorch-geometric dependency, missing a dependency. cannot import name 'DMPNN' from 'deepchem.models.torch_models' (/usr/local/lib/python3.11/dist-packages/deepchem/models/torch_models/__init__.py)\n",
            "WARNING:deepchem.models:Skipped loading modules with pytorch-lightning dependency, missing a dependency. No module named 'lightning'\n",
            "WARNING:deepchem.models:Skipped loading some Jax models, missing a dependency. No module named 'haiku'\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "tasks, datasets, transformers = dc.molnet.load_pdbbind(\n",
        "    featurizer='ECFP',      # 기본: ECFP fingerprint. (GraphConv 등도 가능)\n",
        "    split='random',         # 데이터 분할 방식: random, scaffold 등\n",
        "    subset='core',          # 'core', 'refined', 또는 None (전체). core가 가장 작고 빠름.\n",
        "    reload=True             # 캐시 사용\n",
        ")\n",
        "\n",
        "train_dataset, valid_dataset, test_dataset = datasets"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qGKFDppOd60T",
        "outputId": "0a08c1b2-77b9-4a38-d905-d6cdb873ba08"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "WARNING:deepchem.molnet.load_function.molnet_loader:'split' is deprecated.  Use 'splitter' instead.\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 0, ('/tmp/v2013-core/2d3u/2d3u_ligand.sdf', '/tmp/v2013-core/2d3u/2d3u_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 1, ('/tmp/v2013-core/3cyx/3cyx_ligand.sdf', '/tmp/v2013-core/3cyx/3cyx_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 2, ('/tmp/v2013-core/3uo4/3uo4_ligand.sdf', '/tmp/v2013-core/3uo4/3uo4_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 3, ('/tmp/v2013-core/1p1q/1p1q_ligand.sdf', '/tmp/v2013-core/1p1q/1p1q_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 4, ('/tmp/v2013-core/3ag9/3ag9_ligand.sdf', '/tmp/v2013-core/3ag9/3ag9_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 5, ('/tmp/v2013-core/2wtv/2wtv_ligand.sdf', '/tmp/v2013-core/2wtv/2wtv_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 6, ('/tmp/v2013-core/3dxg/3dxg_ligand.sdf', '/tmp/v2013-core/3dxg/3dxg_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 7, ('/tmp/v2013-core/4tmn/4tmn_ligand.sdf', '/tmp/v2013-core/4tmn/4tmn_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 8, ('/tmp/v2013-core/2zcq/2zcq_ligand.sdf', '/tmp/v2013-core/2zcq/2zcq_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 9, ('/tmp/v2013-core/1q8t/1q8t_ligand.sdf', '/tmp/v2013-core/1q8t/1q8t_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 10, ('/tmp/v2013-core/4des/4des_ligand.sdf', '/tmp/v2013-core/4des/4des_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 11, ('/tmp/v2013-core/1n1m/1n1m_ligand.sdf', '/tmp/v2013-core/1n1m/1n1m_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 12, ('/tmp/v2013-core/1mq6/1mq6_ligand.sdf', '/tmp/v2013-core/1mq6/1mq6_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 13, ('/tmp/v2013-core/2vl4/2vl4_ligand.sdf', '/tmp/v2013-core/2vl4/2vl4_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 14, ('/tmp/v2013-core/2x97/2x97_ligand.sdf', '/tmp/v2013-core/2x97/2x97_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 15, ('/tmp/v2013-core/3k5v/3k5v_ligand.sdf', '/tmp/v2013-core/3k5v/3k5v_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 16, ('/tmp/v2013-core/3jvs/3jvs_ligand.sdf', '/tmp/v2013-core/3jvs/3jvs_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 17, ('/tmp/v2013-core/3fcq/3fcq_ligand.sdf', '/tmp/v2013-core/3fcq/3fcq_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 18, ('/tmp/v2013-core/1kel/1kel_ligand.sdf', '/tmp/v2013-core/1kel/1kel_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 19, ('/tmp/v2013-core/3kgp/3kgp_ligand.sdf', '/tmp/v2013-core/3kgp/3kgp_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 20, ('/tmp/v2013-core/2xy9/2xy9_ligand.sdf', '/tmp/v2013-core/2xy9/2xy9_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 21, ('/tmp/v2013-core/2x00/2x00_ligand.sdf', '/tmp/v2013-core/2x00/2x00_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 22, ('/tmp/v2013-core/2zx6/2zx6_ligand.sdf', '/tmp/v2013-core/2zx6/2zx6_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 23, ('/tmp/v2013-core/3ao4/3ao4_ligand.sdf', '/tmp/v2013-core/3ao4/3ao4_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 24, ('/tmp/v2013-core/2zxd/2zxd_ligand.sdf', '/tmp/v2013-core/2zxd/2zxd_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 25, ('/tmp/v2013-core/3bfu/3bfu_ligand.sdf', '/tmp/v2013-core/3bfu/3bfu_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 26, ('/tmp/v2013-core/2g70/2g70_ligand.sdf', '/tmp/v2013-core/2g70/2g70_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 27, ('/tmp/v2013-core/2j78/2j78_ligand.sdf', '/tmp/v2013-core/2j78/2j78_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 28, ('/tmp/v2013-core/3u9q/3u9q_ligand.sdf', '/tmp/v2013-core/3u9q/3u9q_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 29, ('/tmp/v2013-core/3gbb/3gbb_ligand.sdf', '/tmp/v2013-core/3gbb/3gbb_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 30, ('/tmp/v2013-core/4gid/4gid_ligand.sdf', '/tmp/v2013-core/4gid/4gid_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 31, ('/tmp/v2013-core/3nox/3nox_ligand.sdf', '/tmp/v2013-core/3nox/3nox_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 32, ('/tmp/v2013-core/2pcp/2pcp_ligand.sdf', '/tmp/v2013-core/2pcp/2pcp_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 33, ('/tmp/v2013-core/1w3l/1w3l_ligand.sdf', '/tmp/v2013-core/1w3l/1w3l_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 34, ('/tmp/v2013-core/2vvn/2vvn_ligand.sdf', '/tmp/v2013-core/2vvn/2vvn_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 35, ('/tmp/v2013-core/1qi0/1qi0_ligand.sdf', '/tmp/v2013-core/1qi0/1qi0_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 36, ('/tmp/v2013-core/3bpc/3bpc_ligand.sdf', '/tmp/v2013-core/3bpc/3bpc_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 37, ('/tmp/v2013-core/2vw5/2vw5_ligand.sdf', '/tmp/v2013-core/2vw5/2vw5_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 38, ('/tmp/v2013-core/3muz/3muz_ligand.sdf', '/tmp/v2013-core/3muz/3muz_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 39, ('/tmp/v2013-core/2xhm/2xhm_ligand.sdf', '/tmp/v2013-core/2xhm/2xhm_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 40, ('/tmp/v2013-core/3kwa/3kwa_ligand.sdf', '/tmp/v2013-core/3kwa/3kwa_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 41, ('/tmp/v2013-core/2yge/2yge_ligand.sdf', '/tmp/v2013-core/2yge/2yge_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 42, ('/tmp/v2013-core/4de2/4de2_ligand.sdf', '/tmp/v2013-core/4de2/4de2_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 43, ('/tmp/v2013-core/2wca/2wca_ligand.sdf', '/tmp/v2013-core/2wca/2wca_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 44, ('/tmp/v2013-core/3l7b/3l7b_ligand.sdf', '/tmp/v2013-core/3l7b/3l7b_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 45, ('/tmp/v2013-core/3ozt/3ozt_ligand.sdf', '/tmp/v2013-core/3ozt/3ozt_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 46, ('/tmp/v2013-core/1o5b/1o5b_ligand.sdf', '/tmp/v2013-core/1o5b/1o5b_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 47, ('/tmp/v2013-core/2x8z/2x8z_ligand.sdf', '/tmp/v2013-core/2x8z/2x8z_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 48, ('/tmp/v2013-core/10gs/10gs_ligand.sdf', '/tmp/v2013-core/10gs/10gs_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 49, ('/tmp/v2013-core/3ivg/3ivg_ligand.sdf', '/tmp/v2013-core/3ivg/3ivg_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 50, ('/tmp/v2013-core/3su5/3su5_ligand.sdf', '/tmp/v2013-core/3su5/3su5_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 51, ('/tmp/v2013-core/2xb8/2xb8_ligand.sdf', '/tmp/v2013-core/2xb8/2xb8_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 52, ('/tmp/v2013-core/1nvq/1nvq_ligand.sdf', '/tmp/v2013-core/1nvq/1nvq_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 53, ('/tmp/v2013-core/1oyt/1oyt_ligand.sdf', '/tmp/v2013-core/1oyt/1oyt_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 54, ('/tmp/v2013-core/1sqa/1sqa_ligand.sdf', '/tmp/v2013-core/1sqa/1sqa_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 55, ('/tmp/v2013-core/1hnn/1hnn_ligand.sdf', '/tmp/v2013-core/1hnn/1hnn_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 56, ('/tmp/v2013-core/2xdl/2xdl_ligand.sdf', '/tmp/v2013-core/2xdl/2xdl_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 57, ('/tmp/v2013-core/3udh/3udh_ligand.sdf', '/tmp/v2013-core/3udh/3udh_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 58, ('/tmp/v2013-core/2w66/2w66_ligand.sdf', '/tmp/v2013-core/2w66/2w66_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 59, ('/tmp/v2013-core/3oe5/3oe5_ligand.sdf', '/tmp/v2013-core/3oe5/3oe5_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 60, ('/tmp/v2013-core/2r23/2r23_ligand.sdf', '/tmp/v2013-core/2r23/2r23_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 61, ('/tmp/v2013-core/2qft/2qft_ligand.sdf', '/tmp/v2013-core/2qft/2qft_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 62, ('/tmp/v2013-core/1gpk/1gpk_ligand.sdf', '/tmp/v2013-core/1gpk/1gpk_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 63, ('/tmp/v2013-core/2gss/2gss_ligand.sdf', '/tmp/v2013-core/2gss/2gss_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 64, ('/tmp/v2013-core/3su3/3su3_ligand.sdf', '/tmp/v2013-core/3su3/3su3_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 65, ('/tmp/v2013-core/4djv/4djv_ligand.sdf', '/tmp/v2013-core/4djv/4djv_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 66, ('/tmp/v2013-core/2j62/2j62_ligand.sdf', '/tmp/v2013-core/2j62/2j62_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 67, ('/tmp/v2013-core/1sln/1sln_ligand.sdf', '/tmp/v2013-core/1sln/1sln_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 68, ('/tmp/v2013-core/4djr/4djr_ligand.sdf', '/tmp/v2013-core/4djr/4djr_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 69, ('/tmp/v2013-core/3f3c/3f3c_ligand.sdf', '/tmp/v2013-core/3f3c/3f3c_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 70, ('/tmp/v2013-core/3g0w/3g0w_ligand.sdf', '/tmp/v2013-core/3g0w/3g0w_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 71, ('/tmp/v2013-core/2cbj/2cbj_ligand.sdf', '/tmp/v2013-core/2cbj/2cbj_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 72, ('/tmp/v2013-core/1yc1/1yc1_ligand.sdf', '/tmp/v2013-core/1yc1/1yc1_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 73, ('/tmp/v2013-core/1w4o/1w4o_ligand.sdf', '/tmp/v2013-core/1w4o/1w4o_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 74, ('/tmp/v2013-core/3fk1/3fk1_ligand.sdf', '/tmp/v2013-core/3fk1/3fk1_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 75, ('/tmp/v2013-core/3g2n/3g2n_ligand.sdf', '/tmp/v2013-core/3g2n/3g2n_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 76, ('/tmp/v2013-core/3gy4/3gy4_ligand.sdf', '/tmp/v2013-core/3gy4/3gy4_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 77, ('/tmp/v2013-core/4dew/4dew_ligand.sdf', '/tmp/v2013-core/4dew/4dew_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 78, ('/tmp/v2013-core/3b3w/3b3w_ligand.sdf', '/tmp/v2013-core/3b3w/3b3w_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 79, ('/tmp/v2013-core/2hb1/2hb1_ligand.sdf', '/tmp/v2013-core/2hb1/2hb1_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 80, ('/tmp/v2013-core/2zjw/2zjw_ligand.sdf', '/tmp/v2013-core/2zjw/2zjw_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 81, ('/tmp/v2013-core/1igj/1igj_ligand.sdf', '/tmp/v2013-core/1igj/1igj_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 82, ('/tmp/v2013-core/2brb/2brb_ligand.sdf', '/tmp/v2013-core/2brb/2brb_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 83, ('/tmp/v2013-core/1os0/1os0_ligand.sdf', '/tmp/v2013-core/1os0/1os0_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 84, ('/tmp/v2013-core/1u33/1u33_ligand.sdf', '/tmp/v2013-core/1u33/1u33_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 85, ('/tmp/v2013-core/3utu/3utu_ligand.sdf', '/tmp/v2013-core/3utu/3utu_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 86, ('/tmp/v2013-core/3ejr/3ejr_ligand.sdf', '/tmp/v2013-core/3ejr/3ejr_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 87, ('/tmp/v2013-core/3owj/3owj_ligand.sdf', '/tmp/v2013-core/3owj/3owj_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 88, ('/tmp/v2013-core/3myg/3myg_ligand.sdf', '/tmp/v2013-core/3myg/3myg_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 89, ('/tmp/v2013-core/2weg/2weg_ligand.sdf', '/tmp/v2013-core/2weg/2weg_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 90, ('/tmp/v2013-core/3uri/3uri_ligand.sdf', '/tmp/v2013-core/3uri/3uri_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 91, ('/tmp/v2013-core/3bkk/3bkk_ligand.sdf', '/tmp/v2013-core/3bkk/3bkk_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 92, ('/tmp/v2013-core/3acw/3acw_ligand.sdf', '/tmp/v2013-core/3acw/3acw_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 93, ('/tmp/v2013-core/2qbp/2qbp_ligand.sdf', '/tmp/v2013-core/2qbp/2qbp_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 94, ('/tmp/v2013-core/1e66/1e66_ligand.sdf', '/tmp/v2013-core/1e66/1e66_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 95, ('/tmp/v2013-core/2fvd/2fvd_ligand.sdf', '/tmp/v2013-core/2fvd/2fvd_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 96, ('/tmp/v2013-core/2cet/2cet_ligand.sdf', '/tmp/v2013-core/2cet/2cet_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 97, ('/tmp/v2013-core/1uto/1uto_ligand.sdf', '/tmp/v2013-core/1uto/1uto_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 98, ('/tmp/v2013-core/3mss/3mss_ligand.sdf', '/tmp/v2013-core/3mss/3mss_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 99, ('/tmp/v2013-core/3fv1/3fv1_ligand.sdf', '/tmp/v2013-core/3fv1/3fv1_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 100, ('/tmp/v2013-core/3coy/3coy_ligand.sdf', '/tmp/v2013-core/3coy/3coy_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 101, ('/tmp/v2013-core/1o3f/1o3f_ligand.sdf', '/tmp/v2013-core/1o3f/1o3f_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 102, ('/tmp/v2013-core/1lor/1lor_ligand.sdf', '/tmp/v2013-core/1lor/1lor_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 103, ('/tmp/v2013-core/3vd4/3vd4_ligand.sdf', '/tmp/v2013-core/3vd4/3vd4_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 104, ('/tmp/v2013-core/3pww/3pww_ligand.sdf', '/tmp/v2013-core/3pww/3pww_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 105, ('/tmp/v2013-core/2ymd/2ymd_ligand.sdf', '/tmp/v2013-core/2ymd/2ymd_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 106, ('/tmp/v2013-core/1loq/1loq_ligand.sdf', '/tmp/v2013-core/1loq/1loq_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 107, ('/tmp/v2013-core/2p4y/2p4y_ligand.sdf', '/tmp/v2013-core/2p4y/2p4y_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 108, ('/tmp/v2013-core/3d4z/3d4z_ligand.sdf', '/tmp/v2013-core/3d4z/3d4z_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 109, ('/tmp/v2013-core/2jdm/2jdm_ligand.sdf', '/tmp/v2013-core/2jdm/2jdm_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 110, ('/tmp/v2013-core/3mfv/3mfv_ligand.sdf', '/tmp/v2013-core/3mfv/3mfv_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 111, ('/tmp/v2013-core/3f3a/3f3a_ligand.sdf', '/tmp/v2013-core/3f3a/3f3a_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 112, ('/tmp/v2013-core/1f8d/1f8d_ligand.sdf', '/tmp/v2013-core/1f8d/1f8d_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 113, ('/tmp/v2013-core/1r5y/1r5y_ligand.sdf', '/tmp/v2013-core/1r5y/1r5y_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 114, ('/tmp/v2013-core/3b68/3b68_ligand.sdf', '/tmp/v2013-core/3b68/3b68_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 115, ('/tmp/v2013-core/1vso/1vso_ligand.sdf', '/tmp/v2013-core/1vso/1vso_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 116, ('/tmp/v2013-core/3n86/3n86_ligand.sdf', '/tmp/v2013-core/3n86/3n86_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 117, ('/tmp/v2013-core/3nw9/3nw9_ligand.sdf', '/tmp/v2013-core/3nw9/3nw9_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 118, ('/tmp/v2013-core/3f3e/3f3e_ligand.sdf', '/tmp/v2013-core/3f3e/3f3e_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 119, ('/tmp/v2013-core/3cj2/3cj2_ligand.sdf', '/tmp/v2013-core/3cj2/3cj2_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 120, ('/tmp/v2013-core/1f8b/1f8b_ligand.sdf', '/tmp/v2013-core/1f8b/1f8b_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 121, ('/tmp/v2013-core/3imc/3imc_ligand.sdf', '/tmp/v2013-core/3imc/3imc_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 122, ('/tmp/v2013-core/2y5h/2y5h_ligand.sdf', '/tmp/v2013-core/2y5h/2y5h_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 123, ('/tmp/v2013-core/1a30/1a30_ligand.sdf', '/tmp/v2013-core/1a30/1a30_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 124, ('/tmp/v2013-core/3b3s/3b3s_ligand.sdf', '/tmp/v2013-core/3b3s/3b3s_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 125, ('/tmp/v2013-core/1ps3/1ps3_ligand.sdf', '/tmp/v2013-core/1ps3/1ps3_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 126, ('/tmp/v2013-core/2wbg/2wbg_ligand.sdf', '/tmp/v2013-core/2wbg/2wbg_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 127, ('/tmp/v2013-core/3l4u/3l4u_ligand.sdf', '/tmp/v2013-core/3l4u/3l4u_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 128, ('/tmp/v2013-core/3e93/3e93_ligand.sdf', '/tmp/v2013-core/3e93/3e93_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 129, ('/tmp/v2013-core/2d1o/2d1o_ligand.sdf', '/tmp/v2013-core/2d1o/2d1o_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 130, ('/tmp/v2013-core/1q8u/1q8u_ligand.sdf', '/tmp/v2013-core/1q8u/1q8u_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 131, ('/tmp/v2013-core/3ehy/3ehy_ligand.sdf', '/tmp/v2013-core/3ehy/3ehy_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 132, ('/tmp/v2013-core/3ov1/3ov1_ligand.sdf', '/tmp/v2013-core/3ov1/3ov1_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 133, ('/tmp/v2013-core/4de1/4de1_ligand.sdf', '/tmp/v2013-core/4de1/4de1_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 134, ('/tmp/v2013-core/3nq3/3nq3_ligand.sdf', '/tmp/v2013-core/3nq3/3nq3_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 135, ('/tmp/v2013-core/2qmj/2qmj_ligand.sdf', '/tmp/v2013-core/2qmj/2qmj_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 136, ('/tmp/v2013-core/2xnb/2xnb_ligand.sdf', '/tmp/v2013-core/2xnb/2xnb_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 137, ('/tmp/v2013-core/4g8m/4g8m_ligand.sdf', '/tmp/v2013-core/4g8m/4g8m_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 138, ('/tmp/v2013-core/2xys/2xys_ligand.sdf', '/tmp/v2013-core/2xys/2xys_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 139, ('/tmp/v2013-core/2obf/2obf_ligand.sdf', '/tmp/v2013-core/2obf/2obf_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 140, ('/tmp/v2013-core/3i3b/3i3b_ligand.sdf', '/tmp/v2013-core/3i3b/3i3b_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 141, ('/tmp/v2013-core/1z95/1z95_ligand.sdf', '/tmp/v2013-core/1z95/1z95_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 142, ('/tmp/v2013-core/1w3k/1w3k_ligand.sdf', '/tmp/v2013-core/1w3k/1w3k_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 143, ('/tmp/v2013-core/3dd0/3dd0_ligand.sdf', '/tmp/v2013-core/3dd0/3dd0_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 144, ('/tmp/v2013-core/2vo5/2vo5_ligand.sdf', '/tmp/v2013-core/2vo5/2vo5_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 145, ('/tmp/v2013-core/3pe2/3pe2_ligand.sdf', '/tmp/v2013-core/3pe2/3pe2_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 146, ('/tmp/v2013-core/3kv2/3kv2_ligand.sdf', '/tmp/v2013-core/3kv2/3kv2_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 147, ('/tmp/v2013-core/2xbv/2xbv_ligand.sdf', '/tmp/v2013-core/2xbv/2xbv_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 148, ('/tmp/v2013-core/3pxf/3pxf_ligand.sdf', '/tmp/v2013-core/3pxf/3pxf_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 149, ('/tmp/v2013-core/2yki/2yki_ligand.sdf', '/tmp/v2013-core/2yki/2yki_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 150, ('/tmp/v2013-core/2zwz/2zwz_ligand.sdf', '/tmp/v2013-core/2zwz/2zwz_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 151, ('/tmp/v2013-core/2yfe/2yfe_ligand.sdf', '/tmp/v2013-core/2yfe/2yfe_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 152, ('/tmp/v2013-core/1bcu/1bcu_ligand.sdf', '/tmp/v2013-core/1bcu/1bcu_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 153, ('/tmp/v2013-core/2qbr/2qbr_ligand.sdf', '/tmp/v2013-core/2qbr/2qbr_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 154, ('/tmp/v2013-core/3l3n/3l3n_ligand.sdf', '/tmp/v2013-core/3l3n/3l3n_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 155, ('/tmp/v2013-core/3gnw/3gnw_ligand.sdf', '/tmp/v2013-core/3gnw/3gnw_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 156, ('/tmp/v2013-core/3su2/3su2_ligand.sdf', '/tmp/v2013-core/3su2/3su2_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 157, ('/tmp/v2013-core/3ebp/3ebp_ligand.sdf', '/tmp/v2013-core/3ebp/3ebp_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 158, ('/tmp/v2013-core/3vh9/3vh9_ligand.sdf', '/tmp/v2013-core/3vh9/3vh9_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 159, ('/tmp/v2013-core/1jyq/1jyq_ligand.sdf', '/tmp/v2013-core/1jyq/1jyq_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 160, ('/tmp/v2013-core/1xd0/1xd0_ligand.sdf', '/tmp/v2013-core/1xd0/1xd0_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 161, ('/tmp/v2013-core/2jdy/2jdy_ligand.sdf', '/tmp/v2013-core/2jdy/2jdy_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 162, ('/tmp/v2013-core/1h23/1h23_ligand.sdf', '/tmp/v2013-core/1h23/1h23_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 163, ('/tmp/v2013-core/3l4w/3l4w_ligand.sdf', '/tmp/v2013-core/3l4w/3l4w_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 164, ('/tmp/v2013-core/3ueu/3ueu_ligand.sdf', '/tmp/v2013-core/3ueu/3ueu_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 165, ('/tmp/v2013-core/3g2z/3g2z_ligand.sdf', '/tmp/v2013-core/3g2z/3g2z_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 166, ('/tmp/v2013-core/1lbk/1lbk_ligand.sdf', '/tmp/v2013-core/1lbk/1lbk_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 167, ('/tmp/v2013-core/2zcr/2zcr_ligand.sdf', '/tmp/v2013-core/2zcr/2zcr_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 168, ('/tmp/v2013-core/3lka/3lka_ligand.sdf', '/tmp/v2013-core/3lka/3lka_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 169, ('/tmp/v2013-core/3f80/3f80_ligand.sdf', '/tmp/v2013-core/3f80/3f80_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 170, ('/tmp/v2013-core/3f17/3f17_ligand.sdf', '/tmp/v2013-core/3f17/3f17_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 171, ('/tmp/v2013-core/3ge7/3ge7_ligand.sdf', '/tmp/v2013-core/3ge7/3ge7_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 172, ('/tmp/v2013-core/1hfs/1hfs_ligand.sdf', '/tmp/v2013-core/1hfs/1hfs_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 173, ('/tmp/v2013-core/1zea/1zea_ligand.sdf', '/tmp/v2013-core/1zea/1zea_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 174, ('/tmp/v2013-core/1n2v/1n2v_ligand.sdf', '/tmp/v2013-core/1n2v/1n2v_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 175, ('/tmp/v2013-core/2vot/2vot_ligand.sdf', '/tmp/v2013-core/2vot/2vot_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 176, ('/tmp/v2013-core/1lol/1lol_ligand.sdf', '/tmp/v2013-core/1lol/1lol_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 177, ('/tmp/v2013-core/1f8c/1f8c_ligand.sdf', '/tmp/v2013-core/1f8c/1f8c_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 178, ('/tmp/v2013-core/3n7a/3n7a_ligand.sdf', '/tmp/v2013-core/3n7a/3n7a_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 179, ('/tmp/v2013-core/3gcs/3gcs_ligand.sdf', '/tmp/v2013-core/3gcs/3gcs_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 180, ('/tmp/v2013-core/2ole/2ole_ligand.sdf', '/tmp/v2013-core/2ole/2ole_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 181, ('/tmp/v2013-core/2jdu/2jdu_ligand.sdf', '/tmp/v2013-core/2jdu/2jdu_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 182, ('/tmp/v2013-core/3cft/3cft_ligand.sdf', '/tmp/v2013-core/3cft/3cft_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 183, ('/tmp/v2013-core/3s8o/3s8o_ligand.sdf', '/tmp/v2013-core/3s8o/3s8o_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 184, ('/tmp/v2013-core/2iwx/2iwx_ligand.sdf', '/tmp/v2013-core/2iwx/2iwx_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 185, ('/tmp/v2013-core/2v00/2v00_ligand.sdf', '/tmp/v2013-core/2v00/2v00_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 186, ('/tmp/v2013-core/3huc/3huc_ligand.sdf', '/tmp/v2013-core/3huc/3huc_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 187, ('/tmp/v2013-core/2v7a/2v7a_ligand.sdf', '/tmp/v2013-core/2v7a/2v7a_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 188, ('/tmp/v2013-core/2x0y/2x0y_ligand.sdf', '/tmp/v2013-core/2x0y/2x0y_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 189, ('/tmp/v2013-core/3uex/3uex_ligand.sdf', '/tmp/v2013-core/3uex/3uex_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 190, ('/tmp/v2013-core/2pq9/2pq9_ligand.sdf', '/tmp/v2013-core/2pq9/2pq9_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 191, ('/tmp/v2013-core/1u1b/1u1b_ligand.sdf', '/tmp/v2013-core/1u1b/1u1b_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n",
            "WARNING:deepchem.feat.base_classes:Failed to featurize datapoint 192, ('/tmp/v2013-core/4gqq/4gqq_ligand.sdf', '/tmp/v2013-core/4gqq/4gqq_pocket.pdb'). Appending empty array\n",
            "WARNING:deepchem.feat.base_classes:Exception message: Python argument types in\n",
            "    rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect(tuple, int)\n",
            "did not match C++ signature:\n",
            "    GetMorganFingerprintAsBitVect(RDKit::ROMol mol, unsigned int radius, unsigned int nBits=2048, boost::python::api::object invariants=[], boost::python::api::object fromAtoms=[], bool useChirality=False, bool useBondTypes=True, bool useFeatures=False, boost::python::api::object bitInfo=None, bool includeRedundantEnvironments=False)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "### 구버전 코드\n",
        "```\n",
        "model = cd.models.TensorGraph(batch_size=1000)\n",
        "features = layers.Feature(None, 101, 4)\n",
        "labels = layers.Label(shape=(None, 1))\n",
        "weights = layers.Weights(shape(None, 1))\n",
        "\n",
        "\n",
        "prev = Features\n",
        "for i in range(3):\n",
        "     prev = layers.conv1D(\n",
        "         filters=15, kernel_size=10, activation=tf.nn.relu,\n",
        "         padding = 'same', in_layers=prev)\n",
        "prev = layers.Dropout(dropout=0.5, in_layers=prev)\n",
        "\n",
        "\n",
        "logits = layers.Dense(out_channels=1, in_layers=layers.Flatten(prev))\n",
        "output = layers.Sigmoid(logits)\n",
        "model.add_output(output)\n",
        "\n",
        "\n",
        "loss = layers.sigmoidCrossEntropy(in_layers=[labels, logits])\n",
        "weighted_loss = layers.WeightedError(in_layers=[loss, weights])\n",
        "model.set_loss(weighted_loss)\n",
        "\n",
        "\n",
        "train = dc.data.DiskDataset('train_dataset')\n",
        "valid = dc.data.DiskDataset('valid_dataset')\n",
        "metric = dc.metrics.Metric(dc.metrics.roc_auc_score)\n",
        "for i in range(20):\n",
        "     model.fit(train, nb_epoch=10)\n",
        "     print(model.evaluate(train, [metric]))\n",
        "     print(model.evaluate(valid, [metric]))\n",
        "```\n",
        "\n",
        "\n",
        "---\n",
        "\n",
        "### GPT 이용해 리빌드\n",
        "*   변경점\n",
        "    -   split -> splitter : 공식 인자명 변경\n",
        "    -   roc_auc_score -> rms, mea, r2 score : 가장 최근에 사용한 데이터인 PDBBind 사용  \n",
        "    -   TensorGraph -> tf.keras Functional + dc.models.Kerasmodel : 과거 코드는 tf 1.x버전 기준으로 만들어짐 지금은 tf 2.x버전 이고 호환성, 유지보수등 이제는 사용되지 않는 모델\n",
        "    -   Conv1D 1차원 데이터에서 ECFP 1차원 백터로 변환 : 들어간 데이터가 달라서\n",
        "\n"
      ],
      "metadata": {
        "id": "xc_vc6LvhKrf"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import deepchem as dc\n",
        "from deepchem.models.optimizers import Adam\n",
        "from deepchem.models.losses import L2Loss\n",
        "from tensorflow import keras\n",
        "from tensorflow.keras import layers\n",
        "\n",
        "# 1) 데이터 (회귀)\n",
        "tasks, datasets, transformers = dc.molnet.load_pdbbind(\n",
        "    featurizer='ECFP',\n",
        "    splitter='random',\n",
        "    subset='core',\n",
        "    reload=True\n",
        ")\n",
        "train, valid, test = datasets\n",
        "\n",
        "# 2) 모델: 선형 출력 (활성화 없음)\n",
        "inputs = keras.Input(shape=(train.X.shape[1],), name=\"features\")\n",
        "x = layers.Dense(256, activation=\"relu\")(inputs)\n",
        "x = layers.Dropout(0.5)(x)\n",
        "outputs = layers.Dense(1, name=\"y\")(x)  # 회귀이므로 sigmoid X\n",
        "\n",
        "keras_model = keras.Model(inputs, outputs)\n",
        "\n",
        "model = dc.models.KerasModel(\n",
        "    keras_model,\n",
        "    loss=L2Loss(),                 # 회귀 손실\n",
        "    optimizer=Adam(learning_rate=1e-3),\n",
        "    batch_size=256,\n",
        "    model_dir=\"pdbbind_ecfp_reg\"\n",
        ")\n",
        "\n",
        "# 3) 회귀 메트릭\n",
        "rmse = dc.metrics.Metric(dc.metrics.rms_score, mode=\"regression\")\n",
        "mae  = dc.metrics.Metric(dc.metrics.mae_score, mode=\"regression\")\n",
        "r2   = dc.metrics.Metric(dc.metrics.pearson_r2_score, mode=\"regression\")\n",
        "\n",
        "for i in range(3):\n",
        "    model.fit(train, nb_epoch=3)\n",
        "    print(f\"[{i+1}] train:\", model.evaluate(train, [rmse, mae, r2]))\n",
        "    print(f\"[{i+1}] valid:\", model.evaluate(valid, [rmse, mae, r2]))\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "b6QwsURKeo37",
        "outputId": "805e807a-30a6-4ff7-aae6-627a14a5001e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1] train: {'rms_score': 1.0000000044432944, 'mae_score': 0.815701941714799, 'pearson_r2_score': nan}\n",
            "[1] valid: {'rms_score': 0.9188062058699975, 'mae_score': 0.8020850866051238, 'pearson_r2_score': nan}\n",
            "[2] train: {'rms_score': 1.000000061292397, 'mae_score': 0.8156961704184159, 'pearson_r2_score': nan}\n",
            "[2] valid: {'rms_score': 0.9186259442659855, 'mae_score': 0.8019681419152564, 'pearson_r2_score': nan}\n",
            "[3] train: {'rms_score': 1.0000000945098473, 'mae_score': 0.815706363731144, 'pearson_r2_score': nan}\n",
            "[3] valid: {'rms_score': 0.918944445328111, 'mae_score': 0.8021746906205368, 'pearson_r2_score': nan}\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.11/dist-packages/deepchem/metrics/score_function.py:58: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
            "  return scipy.stats.pearsonr(y, y_pred)[0]**2\n",
            "/usr/local/lib/python3.11/dist-packages/deepchem/metrics/score_function.py:58: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
            "  return scipy.stats.pearsonr(y, y_pred)[0]**2\n",
            "/usr/local/lib/python3.11/dist-packages/deepchem/metrics/score_function.py:58: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
            "  return scipy.stats.pearsonr(y, y_pred)[0]**2\n",
            "/usr/local/lib/python3.11/dist-packages/deepchem/metrics/score_function.py:58: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
            "  return scipy.stats.pearsonr(y, y_pred)[0]**2\n",
            "/usr/local/lib/python3.11/dist-packages/deepchem/metrics/score_function.py:58: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
            "  return scipy.stats.pearsonr(y, y_pred)[0]**2\n",
            "/usr/local/lib/python3.11/dist-packages/deepchem/metrics/score_function.py:58: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
            "  return scipy.stats.pearsonr(y, y_pred)[0]**2\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "###교재 데이터 탐색 후 JunD 결합 여부를 따지는 모델을 만듬\n",
        "-   JunD 데이터 [HCASMC Lonza 0000317155](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1503220&utm_source=chatgpt.com).bed\n",
        "-   염색체 22번 시퀸스 [C22orf31 chromosome 22 open reading frame 31 ](https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=25770).json"
      ],
      "metadata": {
        "id": "JYsCpXiIHwg9"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# Colab 기본: (필요시) 드라이브 마운트\n",
        "# from google.colab import drive\n",
        "# drive.mount('/content/drive')\n",
        "\n",
        "import os, re, gzip, random, json\n",
        "import numpy as np\n",
        "import pandas as pd\n",
        "import tensorflow as tf\n",
        "from tensorflow import keras\n",
        "from tensorflow.keras import layers\n",
        "from sklearn.model_selection import train_test_split\n",
        "from sklearn.metrics import roc_auc_score, average_precision_score\n",
        "from tqdm import tqdm\n",
        "\n",
        "print(\"TF:\", tf.__version__)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yxbiRXMqWFYO",
        "outputId": "2aa0f935-e249-4721-c89a-b1d0c89cfad1"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TF: 2.15.1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# === 입력 파일 경로 (사용자 제공) ===\n",
        "BED_PATH = \"/content/drive/MyDrive/data/JunD/GSM1503220_JUND_PE_q0p0001_top10k_peaks.narrowPeak.bed\"\n",
        "# chr22 관련 NCBI json들은 좌표/서열 추출에 직접 쓰기 어려우므로 참고용\n",
        "NCBI_JSONL = \"/content/drive/MyDrive/data/chromosome 22/ncbi_dataset/data/data_report.jsonl\"\n",
        "NCBI_CATALOG = \"/content/drive/MyDrive/data/chromosome 22/ncbi_dataset/data/dataset_catalog.json\"\n",
        "GENE_FNA = \"/content/drive/MyDrive/data/chromosome 22/ncbi_dataset/data/gene.fna\"  # 보통 전체 chr22가 아님\n",
        "\n",
        "# === chr22 레퍼런스 FASTA 경로 (있으면 지정, 없으면 2단계에서 다운로드) ===\n",
        "CHR22_FASTA = \"/content/chr22.fa.gz\"  # 이 파일이 없으면 2단계에서 자동 다운로드\n",
        "WINDOW = 101  # 101bp, 모델/전처리에서 사용\n",
        "SEED = 1337\n",
        "random.seed(SEED); np.random.seed(SEED); tf.random.set_seed(SEED)\n"
      ],
      "metadata": {
        "id": "pT6xGK6XWMw1"
      },
      "execution_count": 23,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "if not os.path.exists(CHR22_FASTA):\n",
        "    # UCSC hg38의 chr22만 다운로드 (압축 상태로 바로 사용 가능)\n",
        "    !wget -qO /content/chr22.fa.gz http://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr22.fa.gz\n",
        "    print(\"Downloaded chr22 fasta to /content/chr22.fa.gz\")\n",
        "else:\n",
        "    print(\"Found existing FASTA:\", CHR22_FASTA)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9cOPK7HJWWcN",
        "outputId": "16803836-af53-4737-b335-6d4947f75014"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloaded chr22 fasta to /content/chr22.fa.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def load_single_chrom_fasta_gz(path):\n",
        "    \"\"\"gz FASTA에서 chr 전체 서열을 하나의 str로 로드 (UCSC chr22.fa.gz 기준)\"\"\"\n",
        "    seq_parts = []\n",
        "    with gzip.open(path, \"rt\") as f:\n",
        "        for line in f:\n",
        "            if line.startswith(\">\"):\n",
        "                continue\n",
        "            seq_parts.append(line.strip())\n",
        "    seq = \"\".join(seq_parts).upper()\n",
        "    return seq\n",
        "\n",
        "def one_hot_encode(seq):\n",
        "    \"\"\"A,C,G,T만 인코딩, 나머지는 N으로 취급\"\"\"\n",
        "    mapping = {\"A\":0, \"C\":1, \"G\":2, \"T\":3}\n",
        "    arr = np.zeros((len(seq), 4), dtype=np.float32)\n",
        "    for i,ch in enumerate(seq):\n",
        "        j = mapping.get(ch, None)\n",
        "        if j is not None:\n",
        "            arr[i, j] = 1.0\n",
        "        else:\n",
        "            # N 또는 기타 문자는 전부 0\n",
        "            pass\n",
        "    return arr\n",
        "\n",
        "def extract_window(seq, center, window=101):\n",
        "    half = window // 2\n",
        "    start = center - half\n",
        "    end = center + half + 1\n",
        "    if start < 0 or end > len(seq):\n",
        "        return None\n",
        "    return seq[start:end]\n",
        "\n",
        "def read_narrowpeak_as_df(path):\n",
        "    \"\"\"\n",
        "    narrowPeak 형식:\n",
        "    0:chrom, 1:start, 2:end, 3:name, 4:score, 5:strand(.),\n",
        "    6:signalValue, 7:pValue, 8:qValue, 9:peak(정수, summit offset; 없으면 -1)\n",
        "    \"\"\"\n",
        "    cols = [\"chrom\",\"start\",\"end\",\"name\",\"score\",\"strand\",\n",
        "            \"signalValue\",\"pValue\",\"qValue\",\"peak\"]\n",
        "    df = pd.read_csv(path, sep=\"\\t\", header=None, comment=\"#\", names=cols)\n",
        "    return df\n",
        "\n",
        "def filter_chr22(df):\n",
        "    # chr22 표기가 \"chr22\" 또는 \"22\" 등 다양한 경우가 있어 유연하게 필터링\n",
        "    mask = df[\"chrom\"].astype(str).str.replace(\"^chr\",\"\", regex=True).eq(\"22\")\n",
        "    return df[mask].copy()\n"
      ],
      "metadata": {
        "id": "7zozwCmOWXUy"
      },
      "execution_count": 25,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# 4-1) chr22 서열 로드\n",
        "chr22_seq = load_single_chrom_fasta_gz(CHR22_FASTA)\n",
        "chr_len = len(chr22_seq)\n",
        "print(\"chr22 length:\", chr_len)\n",
        "\n",
        "# 4-2) BED 로드 및 chr22 필터\n",
        "bed = read_narrowpeak_as_df(BED_PATH)\n",
        "bed22 = filter_chr22(bed)\n",
        "print(\"chr22 peak rows:\", len(bed22))\n",
        "\n",
        "if len(bed22) == 0:\n",
        "    raise ValueError(\"BED에서 chr22 피크가 없습니다. BED의 chrom 표기(예: chr22/22)를 확인하세요.\")\n",
        "\n",
        "# 4-3) Positive: summit 기준 101bp 추출 (peak 컬럼이 -1이면 구간 중앙 사용)\n",
        "pos_seqs = []\n",
        "pos_centers = []\n",
        "\n",
        "for _,row in bed22.iterrows():\n",
        "    start, end = int(row[\"start\"]), int(row[\"end\"])\n",
        "    peak_off = int(row[\"peak\"]) if not pd.isna(row[\"peak\"]) else -1\n",
        "    if peak_off is not None and peak_off >= 0:\n",
        "        center = start + peak_off\n",
        "    else:\n",
        "        center = (start + end)//2\n",
        "\n",
        "    s = extract_window(chr22_seq, center, WINDOW)\n",
        "    if s is not None and \"N\" not in s:  # 품질을 위해 N 포함 시퀀스는 제외(선택)\n",
        "        pos_seqs.append(s)\n",
        "        pos_centers.append(center)\n",
        "\n",
        "print(\"Positive sequences:\", len(pos_seqs))\n",
        "\n",
        "# 4-4) Negative: 피크 근처를 피해서 임의 샘플링 (동일 개수)\n",
        "# 피크 구간을 확장(exclusion_margin)해서 제외 -> 음성 누수 방지\n",
        "exclusion_margin = 250  # window보다 여유있게\n",
        "exclusions = []\n",
        "for _, row in bed22.iterrows():\n",
        "    start, end = int(row[\"start\"]), int(row[\"end\"])\n",
        "    exclusions.append((max(0, start - exclusion_margin), min(chr_len, end + exclusion_margin)))\n",
        "\n",
        "# 빠른 조회를 위해 머지된 비겹침 interval 생성\n",
        "exclusions.sort()\n",
        "merged = []\n",
        "for st, en in exclusions:\n",
        "    if not merged or st > merged[-1][1]:\n",
        "        merged.append([st, en])\n",
        "    else:\n",
        "        merged[-1][1] = max(merged[-1][1], en)\n",
        "\n",
        "def is_excluded(center):\n",
        "    # 이분 탐색 대신 선형으로도 충분(데이터 적정 규모 가정)\n",
        "    for st, en in merged:\n",
        "        if st <= center <= en:\n",
        "            return True\n",
        "        if center < st:\n",
        "            return False\n",
        "    return False\n",
        "\n",
        "neg_seqs = []\n",
        "tries = 0\n",
        "target = len(pos_seqs)\n",
        "while len(neg_seqs) < target and tries < target * 20:\n",
        "    tries += 1\n",
        "    c = random.randint(WINDOW//2, chr_len - (WINDOW//2) - 1)\n",
        "    if is_excluded(c):\n",
        "        continue\n",
        "    s = extract_window(chr22_seq, c, WINDOW)\n",
        "    if s is None or \"N\" in s:\n",
        "        continue\n",
        "    neg_seqs.append(s)\n",
        "\n",
        "print(\"Negative sequences:\", len(neg_seqs))\n",
        "if len(neg_seqs) < target:\n",
        "    print(\"[경고] 음성 샘플이 양성보다 적습니다. exclusion_margin을 줄이거나 tries를 늘리세요.\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "X3tdbO4cWaKO",
        "outputId": "121dda07-ce7a-481b-842f-f3be648f0501"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr22 length: 50818468\n",
            "chr22 peak rows: 132\n",
            "Positive sequences: 129\n",
            "Negative sequences: 129\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# one-hot\n",
        "X_pos = np.stack([one_hot_encode(s) for s in pos_seqs], axis=0)  # (N_pos, 101, 4)\n",
        "X_neg = np.stack([one_hot_encode(s) for s in neg_seqs], axis=0)  # (N_neg, 101, 4)\n",
        "\n",
        "y_pos = np.ones((len(X_pos),), dtype=np.int64)\n",
        "y_neg = np.zeros((len(X_neg),), dtype=np.int64)\n",
        "\n",
        "X = np.concatenate([X_pos, X_neg], axis=0)\n",
        "y = np.concatenate([y_pos, y_neg], axis=0)\n",
        "\n",
        "# 섞기\n",
        "idx = np.arange(len(X))\n",
        "np.random.shuffle(idx)\n",
        "X = X[idx]; y = y[idx]\n",
        "\n",
        "# train/valid split (층화)\n",
        "X_tr, X_va, y_tr, y_va = train_test_split(X, y, test_size=0.2, random_state=SEED, stratify=y)\n",
        "\n",
        "print(\"Train:\", X_tr.shape, \"Valid:\", X_va.shape)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nR-a8zJ0WdEk",
        "outputId": "a34b29da-6159-4ef0-b277-359ee5f15af2"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Train: (206, 101, 4) Valid: (52, 101, 4)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def build_cnn(window=101, channels=4):\n",
        "    inp = keras.Input(shape=(window, channels))\n",
        "    x = layers.Conv1D(filters=15, kernel_size=10, padding=\"same\", activation=\"relu\")(inp)\n",
        "    x = layers.Conv1D(filters=15, kernel_size=10, padding=\"same\", activation=\"relu\")(x)\n",
        "    x = layers.Conv1D(filters=15, kernel_size=10, padding=\"same\", activation=\"relu\")(x)\n",
        "    x = layers.Dropout(0.5)(x)\n",
        "    x = layers.Flatten()(x)\n",
        "    x = layers.Dense(64, activation=\"relu\")(x)\n",
        "    out = layers.Dense(1, activation=\"sigmoid\")(x)\n",
        "    model = keras.Model(inputs=inp, outputs=out)\n",
        "    model.compile(\n",
        "        optimizer=keras.optimizers.Adam(1e-3),\n",
        "        loss=\"binary_crossentropy\",\n",
        "        metrics=[keras.metrics.AUC(name=\"AUC\"), keras.metrics.AUC(curve=\"PR\", name=\"AUPRC\")]\n",
        "    )\n",
        "    return model\n",
        "\n",
        "model = build_cnn(WINDOW, 4)\n",
        "model.summary()\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bOcTcx6bWenl",
        "outputId": "cd3851d0-0f6e-46f3-f9cd-30f567387114"
      },
      "execution_count": 28,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Model: \"model\"\n",
            "_________________________________________________________________\n",
            " Layer (type)                Output Shape              Param #   \n",
            "=================================================================\n",
            " input_1 (InputLayer)        [(None, 101, 4)]          0         \n",
            "                                                                 \n",
            " conv1d (Conv1D)             (None, 101, 15)           615       \n",
            "                                                                 \n",
            " conv1d_1 (Conv1D)           (None, 101, 15)           2265      \n",
            "                                                                 \n",
            " conv1d_2 (Conv1D)           (None, 101, 15)           2265      \n",
            "                                                                 \n",
            " dropout (Dropout)           (None, 101, 15)           0         \n",
            "                                                                 \n",
            " flatten (Flatten)           (None, 1515)              0         \n",
            "                                                                 \n",
            " dense (Dense)               (None, 64)                97024     \n",
            "                                                                 \n",
            " dense_1 (Dense)             (None, 1)                 65        \n",
            "                                                                 \n",
            "=================================================================\n",
            "Total params: 102234 (399.35 KB)\n",
            "Trainable params: 102234 (399.35 KB)\n",
            "Non-trainable params: 0 (0.00 Byte)\n",
            "_________________________________________________________________\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "callbacks = [\n",
        "    keras.callbacks.EarlyStopping(monitor=\"val_AUPRC\", patience=5, restore_best_weights=True),\n",
        "    keras.callbacks.ReduceLROnPlateau(monitor=\"val_AUPRC\", factor=0.5, patience=2, verbose=1)\n",
        "]\n",
        "\n",
        "history = model.fit(\n",
        "    X_tr, y_tr,\n",
        "    validation_data=(X_va, y_va),\n",
        "    epochs=50,\n",
        "    batch_size=256,\n",
        "    callbacks=callbacks,\n",
        "    verbose=1\n",
        ")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "z2OLYxzSWgsU",
        "outputId": "45874f6d-4e30-41f0-89ec-c34da59edec7"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Epoch 1/50\n",
            "1/1 [==============================] - 3s 3s/step - loss: 0.6970 - AUC: 0.5013 - AUPRC: 0.4764 - val_loss: 0.7058 - val_AUC: 0.3550 - val_AUPRC: 0.4182 - lr: 0.0010\n",
            "Epoch 2/50\n",
            "1/1 [==============================] - 0s 121ms/step - loss: 0.6805 - AUC: 0.6257 - AUPRC: 0.6118 - val_loss: 0.7046 - val_AUC: 0.3565 - val_AUPRC: 0.4262 - lr: 0.0010\n",
            "Epoch 3/50\n",
            "1/1 [==============================] - ETA: 0s - loss: 0.6790 - AUC: 0.6372 - AUPRC: 0.6108\n",
            "Epoch 3: ReduceLROnPlateau reducing learning rate to 0.0005000000237487257.\n",
            "1/1 [==============================] - 0s 124ms/step - loss: 0.6790 - AUC: 0.6372 - AUPRC: 0.6108 - val_loss: 0.7041 - val_AUC: 0.3691 - val_AUPRC: 0.4323 - lr: 0.0010\n",
            "Epoch 4/50\n",
            "1/1 [==============================] - 0s 103ms/step - loss: 0.6710 - AUC: 0.6822 - AUPRC: 0.6955 - val_loss: 0.7038 - val_AUC: 0.3787 - val_AUPRC: 0.4524 - lr: 5.0000e-04\n",
            "Epoch 5/50\n",
            "1/1 [==============================] - ETA: 0s - loss: 0.6648 - AUC: 0.7318 - AUPRC: 0.7169\n",
            "Epoch 5: ReduceLROnPlateau reducing learning rate to 0.0002500000118743628.\n",
            "1/1 [==============================] - 0s 102ms/step - loss: 0.6648 - AUC: 0.7318 - AUPRC: 0.7169 - val_loss: 0.7036 - val_AUC: 0.3757 - val_AUPRC: 0.4448 - lr: 5.0000e-04\n",
            "Epoch 6/50\n",
            "1/1 [==============================] - 0s 113ms/step - loss: 0.6723 - AUC: 0.6711 - AUPRC: 0.6443 - val_loss: 0.7035 - val_AUC: 0.3861 - val_AUPRC: 0.4583 - lr: 2.5000e-04\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# ROC-AUC / PR-AUC\n",
        "proba_va = model.predict(X_va, verbose=0).ravel()\n",
        "roc = roc_auc_score(y_va, proba_va)\n",
        "pr  = average_precision_score(y_va, proba_va)\n",
        "print(f\"Valid ROC-AUC: {roc:.4f} | PR-AUC: {pr:.4f}\")\n",
        "\n",
        "# 임계값 0.5 기준 F1 등 간단 확인 (선택)\n",
        "from sklearn.metrics import f1_score, classification_report\n",
        "pred_va = (proba_va >= 0.5).astype(int)\n",
        "print(\"F1:\", f1_score(y_va, pred_va))\n",
        "print(classification_report(y_va, pred_va, digits=3))\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "N6DLeDfdWil6",
        "outputId": "cd9e42f5-2c77-4177-d105-17a588bf2e17"
      },
      "execution_count": 30,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Valid ROC-AUC: 0.3506 | PR-AUC: 0.4330\n",
            "F1: 0.3673469387755102\n",
            "              precision    recall  f1-score   support\n",
            "\n",
            "           0      0.414     0.462     0.436        26\n",
            "           1      0.391     0.346     0.367        26\n",
            "\n",
            "    accuracy                          0.404        52\n",
            "   macro avg      0.403     0.404     0.402        52\n",
            "weighted avg      0.403     0.404     0.402        52\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "GPT로 만든 test모델은 0.5보다 낮음 -> 모델학습이 재대로 되지 않음"
      ],
      "metadata": {
        "id": "YNTFBybna9Ur"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import json\n",
        "\n",
        "# === 1. JunD narrowPeak ===\n",
        "BED_PATH = \"/content/drive/MyDrive/data/JunD/GSM1503220_JUND_PE_q0p0001_top10k_peaks.narrowPeak.bed\"\n",
        "cols = [\"chrom\",\"start\",\"end\",\"name\",\"score\",\"strand\",\"signalValue\",\"pValue\",\"qValue\",\"peak\"]\n",
        "jund_df = pd.read_csv(BED_PATH, sep=\"\\t\", header=None, names=cols, comment=\"#\")\n",
        "print(\"📌 JunD narrowPeak 미리보기\")\n",
        "display(jund_df.head())\n",
        "\n",
        "# === 2. NCBI data_report.jsonl (JSON Lines) ===\n",
        "JSONL_PATH = \"/content/drive/MyDrive/data/chromosome 22/ncbi_dataset/data/data_report.jsonl\"\n",
        "records = []\n",
        "with open(JSONL_PATH, \"r\") as f:\n",
        "    for line in f:\n",
        "        records.append(json.loads(line))\n",
        "data_report_df = pd.json_normalize(records)\n",
        "print(\"📌 data_report.jsonl 미리보기\")\n",
        "display(data_report_df.head())\n",
        "\n",
        "# === 3. dataset_catalog.json ===\n",
        "CATALOG_PATH = \"/content/drive/MyDrive/data/chromosome 22/ncbi_dataset/data/dataset_catalog.json\"\n",
        "with open(CATALOG_PATH) as f:\n",
        "    catalog = json.load(f)\n",
        "# JSON의 깊은 구조를 표 형태로 보기 쉽게 flatten\n",
        "catalog_df = pd.json_normalize(catalog)\n",
        "print(\"📌 dataset_catalog.json 미리보기\")\n",
        "display(catalog_df.head())\n",
        "\n",
        "# === 4. gene.fna (FASTA) ===\n",
        "FNA_PATH = \"/content/drive/MyDrive/data/chromosome 22/ncbi_dataset/data/gene.fna\"\n",
        "# FASTA는 구조상 한 줄마다 헤더/서열이 교차하므로 테이블로 변환\n",
        "headers, seqs = [], []\n",
        "with open(FNA_PATH) as f:\n",
        "    current_seq = []\n",
        "    current_header = None\n",
        "    for line in f:\n",
        "        line = line.strip()\n",
        "        if line.startswith(\">\"):\n",
        "            if current_header:\n",
        "                seqs.append(\"\".join(current_seq))\n",
        "            current_header = line[1:]\n",
        "            headers.append(current_header)\n",
        "            current_seq = []\n",
        "        else:\n",
        "            current_seq.append(line)\n",
        "    if current_header:\n",
        "        seqs.append(\"\".join(current_seq))\n",
        "gene_df = pd.DataFrame({\"header\": headers, \"sequence\": seqs})\n",
        "print(\"📌 gene.fna 미리보기\")\n",
        "display(gene_df.head())\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 590
        },
        "id": "Q07QCfKibDh_",
        "outputId": "984a5cad-7051-4287-995e-7f182f516f14"
      },
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "📌 JunD narrowPeak 미리보기\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "   chrom      start        end                              name  score  \\\n",
              "0   chr6  141647500  141648043  macs2/JUND_PE_q0p0001_peak_32674   5846   \n",
              "1   chr9  125164615  125165106  macs2/JUND_PE_q0p0001_peak_38902   5775   \n",
              "2  chr11    9587457    9587986   macs2/JUND_PE_q0p0001_peak_5693   5717   \n",
              "3   chrX   30301517   30302156  macs2/JUND_PE_q0p0001_peak_39349   5518   \n",
              "4   chr8   96814070   96814539  macs2/JUND_PE_q0p0001_peak_36555   5518   \n",
              "\n",
              "  strand  signalValue     pValue     qValue  peak  \n",
              "0      .     85.33320  594.12689  584.63629   305  \n",
              "1      .     84.32728  585.95526  577.57861   233  \n",
              "2      .     83.72372  579.83905  571.76343   245  \n",
              "3      .     81.30950  559.53058  551.85291   226  \n",
              "4      .     81.30950  559.53058  551.85291   253  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-78417b64-6f8d-4f06-ba51-26165d963d48\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chrom</th>\n",
              "      <th>start</th>\n",
              "      <th>end</th>\n",
              "      <th>name</th>\n",
              "      <th>score</th>\n",
              "      <th>strand</th>\n",
              "      <th>signalValue</th>\n",
              "      <th>pValue</th>\n",
              "      <th>qValue</th>\n",
              "      <th>peak</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr6</td>\n",
              "      <td>141647500</td>\n",
              "      <td>141648043</td>\n",
              "      <td>macs2/JUND_PE_q0p0001_peak_32674</td>\n",
              "      <td>5846</td>\n",
              "      <td>.</td>\n",
              "      <td>85.33320</td>\n",
              "      <td>594.12689</td>\n",
              "      <td>584.63629</td>\n",
              "      <td>305</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr9</td>\n",
              "      <td>125164615</td>\n",
              "      <td>125165106</td>\n",
              "      <td>macs2/JUND_PE_q0p0001_peak_38902</td>\n",
              "      <td>5775</td>\n",
              "      <td>.</td>\n",
              "      <td>84.32728</td>\n",
              "      <td>585.95526</td>\n",
              "      <td>577.57861</td>\n",
              "      <td>233</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr11</td>\n",
              "      <td>9587457</td>\n",
              "      <td>9587986</td>\n",
              "      <td>macs2/JUND_PE_q0p0001_peak_5693</td>\n",
              "      <td>5717</td>\n",
              "      <td>.</td>\n",
              "      <td>83.72372</td>\n",
              "      <td>579.83905</td>\n",
              "      <td>571.76343</td>\n",
              "      <td>245</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chrX</td>\n",
              "      <td>30301517</td>\n",
              "      <td>30302156</td>\n",
              "      <td>macs2/JUND_PE_q0p0001_peak_39349</td>\n",
              "      <td>5518</td>\n",
              "      <td>.</td>\n",
              "      <td>81.30950</td>\n",
              "      <td>559.53058</td>\n",
              "      <td>551.85291</td>\n",
              "      <td>226</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr8</td>\n",
              "      <td>96814070</td>\n",
              "      <td>96814539</td>\n",
              "      <td>macs2/JUND_PE_q0p0001_peak_36555</td>\n",
              "      <td>5518</td>\n",
              "      <td>.</td>\n",
              "      <td>81.30950</td>\n",
              "      <td>559.53058</td>\n",
              "      <td>551.85291</td>\n",
              "      <td>253</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-78417b64-6f8d-4f06-ba51-26165d963d48')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-78417b64-6f8d-4f06-ba51-26165d963d48 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-78417b64-6f8d-4f06-ba51-26165d963d48');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-9b8cd7b1-3bc6-4e17-a562-d56801665184\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-9b8cd7b1-3bc6-4e17-a562-d56801665184')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-9b8cd7b1-3bc6-4e17-a562-d56801665184 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"display(gene_df\",\n  \"rows\": 5,\n  \"fields\": [\n    {\n      \"column\": \"chrom\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"chr9\",\n          \"chr8\",\n          \"chr11\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"start\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 58198719,\n        \"min\": 9587457,\n        \"max\": 141647500,\n        \"num_unique_values\": 5,\n        \"samples\": [\n          125164615,\n          96814070,\n          9587457\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"end\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 58198687,\n        \"min\": 9587986,\n        \"max\": 141648043,\n        \"num_unique_values\": 5,\n        \"samples\": [\n          125165106,\n          96814539,\n          9587986\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"name\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"macs2/JUND_PE_q0p0001_peak_38902\",\n          \"macs2/JUND_PE_q0p0001_peak_36555\",\n          \"macs2/JUND_PE_q0p0001_peak_5693\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"score\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 150,\n        \"min\": 5518,\n        \"max\": 5846,\n        \"num_unique_values\": 4,\n        \"samples\": [\n          5775,\n          5518,\n          5846\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"strand\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \".\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"signalValue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1.819585359965288,\n        \"min\": 81.3095,\n        \"max\": 85.3332,\n        \"num_unique_values\": 4,\n        \"samples\": [\n          84.32728\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"pValue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 15.690013700448771,\n        \"min\": 559.53058,\n        \"max\": 594.12689,\n        \"num_unique_values\": 4,\n        \"samples\": [\n          585.95526\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"qValue\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 15.02550929713203,\n        \"min\": 551.85291,\n        \"max\": 584.63629,\n        \"num_unique_values\": 4,\n        \"samples\": [\n          577.57861\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"peak\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 31,\n        \"min\": 226,\n        \"max\": 305,\n        \"num_unique_values\": 5,\n        \"samples\": [\n          233\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "📌 data_report.jsonl 미리보기\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "                                         annotations chromosomes commonName  \\\n",
              "0  [{'annotationName': 'GCF_000001405.40-RS_2025_...        [22]      human   \n",
              "\n",
              "                           description     ensemblGeneIds  \\\n",
              "0  chromosome 22 open reading frame 31  [ENSG00000100249]   \n",
              "\n",
              "                                     geneGroups geneId orientation  \\\n",
              "0  [{'id': '25770', 'method': 'NCBI Ortholog'}]  25770       minus   \n",
              "\n",
              "   proteinCount swissProtAccessions    symbol               synonyms taxId  \\\n",
              "0             6            [O95567]  C22orf31  [HS747E2A, bK747E2.1]  9606   \n",
              "\n",
              "        taxname  transcriptCount  \\\n",
              "0  Homo sapiens                6   \n",
              "\n",
              "                                transcriptTypeCounts            type  \\\n",
              "0  [{'count': 2, 'type': 'PROTEIN_CODING'}, {'cou...  PROTEIN_CODING   \n",
              "\n",
              "  nomenclatureAuthority.authority nomenclatureAuthority.identifier  \n",
              "0                            HGNC                       HGNC:26931  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-0a2c679e-e067-400e-933d-9169378a92a6\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>annotations</th>\n",
              "      <th>chromosomes</th>\n",
              "      <th>commonName</th>\n",
              "      <th>description</th>\n",
              "      <th>ensemblGeneIds</th>\n",
              "      <th>geneGroups</th>\n",
              "      <th>geneId</th>\n",
              "      <th>orientation</th>\n",
              "      <th>proteinCount</th>\n",
              "      <th>swissProtAccessions</th>\n",
              "      <th>symbol</th>\n",
              "      <th>synonyms</th>\n",
              "      <th>taxId</th>\n",
              "      <th>taxname</th>\n",
              "      <th>transcriptCount</th>\n",
              "      <th>transcriptTypeCounts</th>\n",
              "      <th>type</th>\n",
              "      <th>nomenclatureAuthority.authority</th>\n",
              "      <th>nomenclatureAuthority.identifier</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>[{'annotationName': 'GCF_000001405.40-RS_2025_...</td>\n",
              "      <td>[22]</td>\n",
              "      <td>human</td>\n",
              "      <td>chromosome 22 open reading frame 31</td>\n",
              "      <td>[ENSG00000100249]</td>\n",
              "      <td>[{'id': '25770', 'method': 'NCBI Ortholog'}]</td>\n",
              "      <td>25770</td>\n",
              "      <td>minus</td>\n",
              "      <td>6</td>\n",
              "      <td>[O95567]</td>\n",
              "      <td>C22orf31</td>\n",
              "      <td>[HS747E2A, bK747E2.1]</td>\n",
              "      <td>9606</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>6</td>\n",
              "      <td>[{'count': 2, 'type': 'PROTEIN_CODING'}, {'cou...</td>\n",
              "      <td>PROTEIN_CODING</td>\n",
              "      <td>HGNC</td>\n",
              "      <td>HGNC:26931</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-0a2c679e-e067-400e-933d-9169378a92a6')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-0a2c679e-e067-400e-933d-9169378a92a6 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-0a2c679e-e067-400e-933d-9169378a92a6');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"display(gene_df\",\n  \"rows\": 1,\n  \"fields\": [\n    {\n      \"column\": \"annotations\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"chromosomes\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"commonName\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"human\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"description\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"chromosome 22 open reading frame 31\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ensemblGeneIds\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"geneGroups\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"geneId\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"25770\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"orientation\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"minus\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"proteinCount\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 6,\n        \"max\": 6,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          6\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"swissProtAccessions\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"symbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"C22orf31\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"synonyms\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"taxId\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"9606\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"taxname\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"Homo sapiens\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"transcriptCount\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": null,\n        \"min\": 6,\n        \"max\": 6,\n        \"num_unique_values\": 1,\n        \"samples\": [\n          6\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"transcriptTypeCounts\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"type\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"PROTEIN_CODING\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"nomenclatureAuthority.authority\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"HGNC\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"nomenclatureAuthority.identifier\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"HGNC:26931\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "📌 dataset_catalog.json 미리보기\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "  apiVersion                                        genes.files\n",
              "0         V2  [{'filePath': 'gene.fna', 'fileType': 'GENOMIC..."
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-e3162181-7200-4c4e-a444-98bbeea29de2\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>apiVersion</th>\n",
              "      <th>genes.files</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>V2</td>\n",
              "      <td>[{'filePath': 'gene.fna', 'fileType': 'GENOMIC...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-e3162181-7200-4c4e-a444-98bbeea29de2')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-e3162181-7200-4c4e-a444-98bbeea29de2 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-e3162181-7200-4c4e-a444-98bbeea29de2');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"display(gene_df\",\n  \"rows\": 1,\n  \"fields\": [\n    {\n      \"column\": \"apiVersion\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"V2\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"genes.files\",\n      \"properties\": {\n        \"dtype\": \"object\",\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "📌 gene.fna 미리보기\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "                                              header  \\\n",
              "0  NC_000022.11:c29073943-29058672 C22orf31 [orga...   \n",
              "1  NC_060946.1:c29535171-29519948 C22orf31 [organ...   \n",
              "\n",
              "                                            sequence  \n",
              "0  GATGTCGCTGGAGGTCTCGGAGAGGGGCGTATAGCCTCTTGTAGCG...  \n",
              "1  GATGTCGCTGGAGGTCTCGGAGAGGGGCGTATAGCCTCTTGTAGCG...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-84dc5715-5bf5-4b3f-a5f6-d7d4528463c9\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>header</th>\n",
              "      <th>sequence</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>NC_000022.11:c29073943-29058672 C22orf31 [orga...</td>\n",
              "      <td>GATGTCGCTGGAGGTCTCGGAGAGGGGCGTATAGCCTCTTGTAGCG...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>NC_060946.1:c29535171-29519948 C22orf31 [organ...</td>\n",
              "      <td>GATGTCGCTGGAGGTCTCGGAGAGGGGCGTATAGCCTCTTGTAGCG...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-84dc5715-5bf5-4b3f-a5f6-d7d4528463c9')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-84dc5715-5bf5-4b3f-a5f6-d7d4528463c9 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-84dc5715-5bf5-4b3f-a5f6-d7d4528463c9');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-8a102016-ffa3-4df6-8dc7-064262be2217\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-8a102016-ffa3-4df6-8dc7-064262be2217')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-8a102016-ffa3-4df6-8dc7-064262be2217 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"display(gene_df\",\n  \"rows\": 2,\n  \"fields\": [\n    {\n      \"column\": \"header\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"NC_060946.1:c29535171-29519948 C22orf31 [organism=Homo sapiens] [GeneID=25770] [chromosome=22]\",\n          \"NC_000022.11:c29073943-29058672 C22orf31 [organism=Homo sapiens] [GeneID=25770] [chromosome=22]\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"sequence\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"GATGTCGCTGGAGGTCTCGGAGAGGGGCGTATAGCCTCTTGTAGCGGGGGAGTCGTCCCGGTGCAAGGTGCGTCGATGAAGGACTTGGGGCGGGATCCCAGGAGGTACTGGGGAGCAGGGGTCGTCCCGGCCCGGGAGGGGTCGCTGGGGACTTGGGGACGGGATCCCAGGAGGTACCGGGAAGCAGGGGGCCTCCTGGCCCGTGGGGAGTCGCTGGGGATTTTGGGACGGGATCCTAGGAGGTACCGGGGAGCAGAGGACGTCCCGGCCCGGGGGAAGTCGCTGGGGACTTGGGACGGGATCCCAGGAGGTACCGGGGAGAGGGGGCGTCCCGGCCTGGTTGGGGTCGCTGAGGGACTCGGGCAGCGGGTCCCAGATCCCGAGACAGAGAAGATGTCCCAGCCCGGGGGGCGTCGCAGAGGGTCTTGGAAGACTCAAGGCGGAGAGGGCCTCCTGGGCTGAGGGGCGTCGCTGAGGAGCACCGGGTCCTGCAGCCGGGGGACAGTACCGGGCCCGGGGGCGTCGATGACGGGCCTGGGGGTAGCCCGGGTCCCAGGAAGGGACCATCCCGGCCTGAGGGGCGTCGCCTAGGGGCGAGGGGAAGGGGCCCGGAAGGTCCTGAAGCGGGAGGGAGGGGGCTCTCAGGCCGGCCGGCCCTTGTCGCCCCTCGAGTGGGCTCCGCTCAGGGCGGGCGGCGGGGGGTCGCTCACACTCACCGGGTCCGGGGCCGAGGCCGGGGCTAGGCGCGGGCCGGGCCGCCAGCGTGAGCGCCGCGGCGGAGAGCAGGGCGAGGCGGGCGGCGGGCGGCGCCATGGGCCGTCAGTGCGGGGCCCCGGGGCGCGGCCCGGGGTAAAGGGGGCAGCGCGGGGAGCCGGGGGCGGCCATGGGAGCAGGACGCGGGGCCCGAGTGTAGCCCCGCCCCGGCCGCCCGGCCGCCCCGCCCCGGCCCGCCCAGCTCAGCGCCGCCACCGCCCCTGCAGCCTGGGGCCCGCCCGGATCCGCGGAGAGGCCCGGGGGGCGGAGCCCACTGCGCACGCACGCCCCGCCCCGGCCGCGCGCTGCTCCTGCCACTCACGCATCAGGCCCCGCCCATCTCTCGCCTCACCTGGGCCGGCCCAGCAGGGCCCTCTGTTTGTCCCATCCAAGGGTAGGATCAGCAGGTGAGGGGCGGGGCCTCGGCCCGGCCTCCAAGTCACCAAGCGGTTTGGCCGGGCTGAAATAGGCCGGGCTCCCAGTGACCCCTGCTCCACCCTCACTTAGGTCGTGACCCACGCGCGAGCTCAGCTACAGCGGTAATAATTACAACAGCAGTTTGCAAACTGTTTTCCCTTGCTTAGCTCATCGTAATGGTTAAAAAAAGAATGCTTTGGAGAGTTCACATCCCACTTGGTCACATGTAACACCTTGGACAGGTTTAAAGGAAAAATAGCTCTTACCTCAATTTCCTCATCTGTTAAATGGAGGTGATAATAGCTACTACTCCTCAGAGCGCTTGCACTAAATAAGAATGCCTACAGCCGAGTGCGGTGGCTAACAACTCCTGTGATCCCAGCACTTTGGGAGGCCGAGACGGAAGAATCGCTTGAGCCCAGGATTCTGAGACCAACCTGGGCAACACAGCGAGACCCCATCTCTACAAGAAAAAAATAAATAACTCAGTGGTCCCAGCTACTCAGGAGACTGAAGCAGGAGGATCGCTTGAGCCCTGGAGGTCGGGCTTGCAGTGAGCCTTCTTCGCACCACTGCACTCCAGCCTGGGTGACAGAGCAAGACCCTATCTCAAAAACAAAAACAAAAACAAAAAAAAACCAAAACGAAAGCATGGAAGCTTTTACGCACGCAGAGTAGGAGTTCAGCAAATGTTGAATTTCTTTCTCCACCCCAAGTTTGCAGCCACATACTGACCCCTGGTTCCGTATCTTATCACATAGCCAGTTTCAACCAGGGCAACTTTCCTGATCTGTCTGAGGCTCAGCCGTTTGCAGAATAGGAATAATAAAAGGCCTTCAGAGGATTCGATGAGATAGAAGCAAGTAAAACGCCCAACGCAACGTCCAGAAATGGATGAGACAGTACTGCTTATCCTACAGCCTGGGATGTTAGGGATCAGAGAGCGCGCGGCTTGCTTAAACTTTCGGCCCCAGCGTGGTAGGCCCTTTGCCCGCCTCCCCCCGCCCCGACCCCGGCAAGGTCTTCCCCTGCCGCACCCGGGGTCGGTCGTGGGCCGTCCTCCTCCCCCGCCCTCCCGCCAGGCAGAGGGGATTTTCCTACAGTTCCCGCCTAAAGCGCAGCGGTGACGCCCGGCGGGGCAGGTCCAGGCGGCGCGCCGGGCGCTTTCAAGTCCCTGGTCCCGGGCTTGCAAGTGCAGACCTGCGCCCGGCGCGGCGGCGGGAACATCTGCACCAGTGTGCAGACTCGCCGCACCCAGAGCGCGGCTGGGCGGGACCGGCACGCCTTGGGTACACTCGCCCCTGGGCCCCCTCAACCCCCGGGCGCCACCCCCTTATCCACTCCCAGAGCCCGCGTACCCCCAAACTCCTTTCCCCACCCACGGCTGCTCGCAGGGCCGGGCCCCAAGGGCGCAGCTACCTCCCACGAGGCCTCTCGATCTCTGTCCCTCTCCACCCACACGACTCAGCGCTGTAGCAAGAAGAATCTTGTTCAGTTATTGTTTTCATGTGTCATTCCCGAGTTCAAGGATCCTCTGGGCTTCTTATTGGAACTTAGGGGCACCCACCATCCCCTCACCCAGCCAGCCTCGGTCCTGGGGACCGCCCCCCCCACCTCCTGCCACCGGTGATCCCCACTTTTGTCCAGCTTTCCCCTTTTCCACAATTTTGGGCCCTTCCCACCTTCTCCACCGGACCTGAGAACCCTTATTCCAGGACTTCTGTGCCCTCCCAGACAGCAGTGCCTTTGAAGAGAATGAAATGAGAAGATGCAGGTGAAGCTCTCAGCCCAAAGCAAACACTCAATAAAGTCGAACTTTCCTTCGTTCAGCAAATATTCAGCGAGGAACATGCCAAGGGCTGAGGATACAGCCACCAAGACCAAACCTTCCTCTTGGAACTTACATTCTAGTCAACCTGAAAACACGCAAATTCCGGAGAACAAAAAGTACTGCAAAGACAAGAAAATTATGTTGACCACATGTTCTGTGGATTTTTTTTCTTTTTCTTTTTCTGGAGACAGGGCCTCCGTCTGTCGCCCACACTGGAGTGCAGTGGTGAGATCATGGCTCACTGCAGCCTCCAATTCCTGGGCTCCAGTGATCCTCCCGCTTCAGCCTCCTGGATAGCTGGGACTACAGAGGCATGCCACCACACCTGGCTAACCTTTGGACTACTTTAAGCAGGATGCCAGAGAAAGACTTCAGCAAAGCAGGACACAGGTCTCTCCAGTTCACAAGGTTCTTTCTGCATCTCATTCAGGTTTCAGAGGAGAAAATAGGAAGGGCACCCAGGAGGTTGGAAACAAAGTTTATTTGGCCAAGTCTGGGCAGCCATGCCTTGTGATAGTAGCTTTGGTCTGTCCTGCCTGCTGCGGGAAGGCACCCCTCCTGGGCTGGCAGCAGCCCCTCCTCCAGTTGGCTCTCTCTGTCATGTACACTGCTTAGGACCAGCCATATAATTGGCATGACACAGTGAGAAATGAAAATGTATGTCCTTTGTCCAAACTTATTAAGAATTTCAAGACAGGCCAGGCGCTGTGGCTCACGCCTGTAATCCCAGCATTTAGGGAGGCCAAGGCAGGTGGATCACCTGAGTTCAGGAGTTTGAGACCAGCCTGGCCAACATGGTGACACCTCATCTCTACTAAAAACACAAAAAATTATCCGGGCTTGGCGGTAGGTGCCTGTAATCCCAGCTACTCAGGAGGCTGAGACAGGAGAATTGCTTGAACCTGGGAGGTGGAGGTTGCAGTGAGCCAAGATCGTGCCACTGCACTCCAGCCTGAGCGACAGAGCAAGACTCCATCTCAAAAAAAAAAAAAAAAAAAGAATTTCAAGACAGTGACAGCAGAAGATGAAAAGATAAAGCATGTGTGGGGCCCTTCCAAGCGAGGGGGCCTATGTTGATATACAGGTTGCACGCCCATGAAAATGGTCCTATTGTACAACAGGACTCAGAAATTGTGTGCAAGCTCCTCCTGGCACTGGCTGCAGTGAAAAACATCCCTCTAACCTTCTCCACCCCCACCTTCTCTTTTCCAGCTGAAAGCCCAGCACAGGTGATGGAAACAGTGCACAAGAAGCTAAAAATACAGCCCAGCACGTCTAGGTCATGGCACTTAACAGCTCAAGGTCTTTTGTGGTCCTGGCTTCTGGCAGAGAGCAACCCGGGATACTCATTTCAGTGCTTTACATGCCTGGGGTGTTTAGCAGCTTTATTTCATGGAAAGAGTGAGGACACTGAATGCAGCTCCTGGGCCTCCAGAAAGATGTTGGTGTCCGTTTGTCTGTTTGCACTGTGCAAGTAAGGCCCCTTTCTTCCTCCGAGGGCAGCAGGATTGGCCAAGAAGCCAGGAGACCTACAATGTTGGAGTTTGGGTCATTCAACTGTCTTTGAAACTCCAGTTGGGATGATTTCTGCTCAAAGGGCAGTCATTTTATGGAAAGAGCCCTGGAGGAGGGGGCCTGATTATAATGACAGCAATAGCAAATGTTTACTGCAGACTTGACTAAGGCTATGCAGGGTGTCTACGGTTGCATTTCACAAAATCCGCCCAGTAGCCTTATGAGGGAGGCATTCTTACCAGCCCCACACCCACCCCAGAACAAAGAGAGATTAAATGACTTGCCCAAAGGTGACATCGTGTTAGAGCTGGGCTTTGAACCCAGGCAGTGGAACCCTAGTCTTACCCTCACAAGTACCAGGCTGGGCCAGGGGTGGTGGCTCACACCTGTAATCCCAGCACTTTGGGAGGCCGAGGCAAGAGGATCACAAGGTCAGGAGTTCAAGACCAGCCTGACCAACATAGTGAAACCCCATCTCTACAAAAATACAAAAACTAGCCAGGTGTAGTGGCGGGCGCCTGTAGTCTCAGGTACTTGGGAGACTGAGGCAGGAGAATGGCTTGAACCCAGGAGACAGAGGTTGCAGTGAGCCGAGATCGTGTCACTGCCCTCCAGCCTGGGAGACAGACCAAGACTCCATCTCAAAAAAAAAAAAAAAAAGTACCAGGTTGAGCCACCTCCTTGTCTTTTTTGTTTTAAAGACAGGGTCTTGCCGGGCACGGTGGCTTACGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGTGGGCAGACCATGAGGTCAGGAGATCGAGACTATCCTGGCTAACATGGTGAAACCCCGTCTCTACTAAAAATACAAAAAATTAGCCAGGCATGGTGGCAGGTGCCTGTAGTCCCAGCTACTCAGGAGGCTGAGGCAGGAGAACGGTGTGAACCCGAGAGGCAGAGGTTGCAGTGAGCCAAGATCACACTACTGCACTCCAGCCCAGGCAACAGAGCAAGACTCCGTCTCAAAAAAAAAAAAAAAGAAATTTAAAGACAGGGTCTTGAGCCCGGGCAGGATGGCTCGTGCCTGTAATCTCAGCACTTTGGGAGGCTGAAGCGGGTGGATCACTTGAGGTTAGGAGTTCGAGACCAGCCTGGCCAACATGGTGAAACCCCATCTGTAAAAAATAAATAAATAAATAAAAATTAGCCAGGTGTCGGGGCATGCGCCTATAATGTCAGCTACTTGGGAGGCTGAGGCAGGAGAATCACTTGAACCCGGGAGACAGAAGTTGTAGTGAGCAGAGATAGTGCCATTGTCCTCCAACCCAGGCAAGAGTGAGACTCTGTCTCAAAAAAAAAAAAAAAAAAAGACGGGGTTTGATATGTTGCCCAGATTGGATTGCAGTGACTATTCACAATCATGGCACACTACAGCCTCGAATTCCTGGGCTCCAGCAATCCTCTCACCTCAGACTCAGTAGCTGGGGCTACAGATGCTTGCTACTGTACCTGATGCCTCCTTATTCTGTCATTAGGCCAGGTATGGTGGCTCACATTTGTAATCCCAGTACCTCGGGAAGCTGAGGTGGGAGGATCTCTTGAACCCAGTATTTTGAGGTTACAGTGAACGAGGATACAGCCACTGCAACCCAGCCTGGGCAACAAGAAAGCCAGACCCTGTCTCTAAAATTAATAATAATAGCCTGGGCACGGTGGCTCATGGTTGTAATCTCAGTACCTTGGGAGTCCGATGCTGGCAGATCACCTGAGGTCAGGAGTTCGAGTCCAGCCTGGCCAATATGGCAAAACCCCAGTTCTGCTAAAAATACAAAAATTAGCCGGGCGTGGTGGCTCATGCCTGTAATCCCAGCTACTTGGAAGGCTAACACAGGGTTCAGCTACTTGAACCCTGGTGGTGGAAGTTGCAGTGAGCCAAGGTCATGTCACTGCACTCTAGCCTGGGCGAGAAAGCGAGAGTCTGTGTCAAAAAAGAAAGAAAAGAGAAGAAAAAAGAAAATTAATAATAATAATTAATAAAAAATAAAATAATAAATTTCTGTCACCAAATGCTTACTGAGTGGTCCAGAGCTGGGACTGGAATGGGGCAAGTGAAGTATGTGCCTCAGGCACAGAGTTTAAGAGGATTTTTAGAAAACACAATAAAAAATCCAGTAATCAAGATAAATATTATTTAATAGATATTTTATACATCTAATAAATATATTTAGTAAATAGTATTTAATGCAATATTTTAAAATAATTTTTTAACATCTAAATGAATGCAAAAAAATCCATTATGAACAGACTGTCCCAGCTTTAAATGAAGATAGGATCTGTATTCCCAATTTTCCTTCAGCCTCGGGTTTGAGGCTGCACTGTATCATTACTGAGTCTCCCAGGATTCTGGCTCAGACAACTAGTCATGGGAAATTCAAGAGAGGCAGTCATTTGCTTGTTCTGATGGGGGTGAAGGAAGAGAAAGACTCTGGTCTTGAACATAATGAGTTAGAGTTGTCTGTGGGGCATCCAGGTGGGCACCTCTGAAGACTCACAGTAGGGTTTGGGTTAGCATTGAGGAGGCGTTGGCTGGGCGCGGTGGCTCCCGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGCAGGTGGATCACCTGAGGTCAGGAGTCCAAGACCAGCCTGGCTGACATGGTGAAACCCTGTCTCTACTAAAAATACAAAAATTAGCTGGGCATGGTGGTGAGTGCTTGTAATCCCAGTTACTCAGTAGGCTGAGGCAGGAGAATTGCTTGAACCCAGGAGGCGGAGGTTGCAGTGAGCCAAGATCGTGCCACTGCACTCCACCCTTGGCAGCAGAGCGAGACTGCATCTCAAAAAAAAAAAAAAAAAAAAAAAAAAGAAAAGAAAAGAAAAGAAAGAAAGAAAGAAAGGAGGAAGTGTGGGACATCTGCGGGAGTCAGGTGAGAACTGGTGGGAAGGCACACAGGACACCATTATTTTGCATACTTGGCCCATGATTGGTGCATATTCAGCACATGTGGTCCAAATCATTGGTCCTCCGAGTTTGCATCACAGTTCAGGAATCTTTTCCAATTCTTTCCCATCCAAAGTGTTTCTCTTTTGCATCATGAACCTCAGATATGTTCCCACATGGGTATGGAAGTTTACTGCTTGGGTGTTAAAACAAAAACATCTGAGGAAATTAAAAGCAACTAAATTCCACTAGAATTGCATGGGTGGGGCAGGGCCATAAAGGTGGAAGACAAGAATTAATAGAAATTGACTCAATTTCTTCTTTATTAGCTGATCCACCAGAAGGAGAATAAGAGCTAAGAGAAGCTAAAAAGAAGAAAGGTCGTAGTAAAGGAAAGACAAAAAAAAAAGGAAGAAGGAAGATTATACGAGCTAAAAATACTGTGAGGAAAGTTTGGCTAAACATCAAAGAAGGACAGATGTGTTTGCAAATAGAAGCGGTGGTTTAGATATGTCAGGAAGCACAGCTCCAAATGATGGGTGGAAAATAAAGGTTAGTGAGACTGAGCTCAGCAGAATAAAAAAGCTTCATGGGAGAATCAAACTAAGGAAGAGGTCTTTCAGCAACATGGAAGCCCTACTGCTTCAACCCCGAGTTCCCCGGATCAAGTGCTGGCACCCATGATGGAAACTCTTGCCATGGTTTTAGTACCCTGGACCAAGTAGTCATTCCATCCTGACTTTAAAATTCTAAACAGCCTTTGATGGGACAATCTCTGCTAAAGACTAACCACTTCCTTATCTTATCTTCAGCTACCTGCTTCCCTTTCTGTTTAACAAAGCATAGAATATTCTGAACAACTCCCGTAACCAGGGGATGTAGCTCCTCTACATTCAACACTTCTGTGAGGTCAATCGATCTGACAAATATATAATGAAGATGGATCATGAAAGTCTGATACCCAAGTCAGATCAGAAACTTTCCCTACTTCCAAGAGGCTCATTGCTTGGCCTCATACACTAGATTTTTTTTCTTTGTTCTTTTTCTTTTTTGTAGAGATGGGGTCTTGCTTTGTTGCCCAAGCTCATCTTGAACTCCTGGATTCAAGTGATCCTTCTGCCTCAGCCTCCCAAAGTGCTGGGATTATAGGTGCGAGCCACCACACCGGCCATAAACCAGATTGATGTACATGAATGACCTGAGACAAAGATCATATCAAATGTTGCATTACATCACTTAGACCATGAGCAACAGGAGCTCAAGAAAGGGTCAGATGAGTAAATGGACCATAATATTAATATCATTCACATGGTCAGGCGCAGTGGCTCACGCCTGAAATCCCAGCACTTTGGGAGGTGAAGGCTGGTAGATCACGAGGTCAGGAGTTCGAGACCAGCCTGGCCAACATGGTGAAACCCCCATCTCTGCTAAAAATACAAAAATTAGCCGGGCGTGGTGGTGCACACCTGTAATCTCAGCTACTCGGGAGGCTGAGGCAGGAGAATTGCTTGAACCCAGGAGGCGGAGGTTGCAGTGAGCTGAGAATGTGCCATTGCACTTCAGCCTGGGTGACAGATTGGGACTCCATCTCAAAAAAATAAATAAGTAAAAAATAACTTTCACATGGGTTAGGTGTGTTGGTTCACAGCTGTAGTCCCAGCACTTCGGGAGGCCAAGGAGGATCACTTGAGACCAGGAGTTCCAGACTAAGCTGAGCAACACAGGGAGACCTCATCTCTACAAAAAAAAAAAAAAAAAAAAAAAAAATCACTGGGCAAAAACAAGCCTGTGATCCCAGCTACTTGGGAGGCCAAAGTGGGAGGATTGCTTGAGCCCAGGAGGTCGAGGCTCTCATGAGCCGTGATCACACCACTGCACTCCAGCCTGGGTGACAGACTGAGACCCTGTCTTGAAACAAAAACAAAAACACAAAAAAGCAAATATTCATTGAGTGCTTACTATATACACCAGGAGCCATGCCAAGTATTTCAGATGCATTGTCCCACTTCAAACTCACAATAACCCTTAGGTAGGCAGTATGACTAGGCCCATTTCACAGCTGAGGAAATTAAGGTTCAGGGGTATGAATTCACTTGCCTAATGAAAGATGGATCTGGGATGCCATCACAGGCAGATTCCCTCCTGGGCCTAGAGAAAGTCTCATGACAAGGACAGGACTAGAACTGGGCCCGGGTATGGTAGATGGATGGGCTGACCAACAGAGAGGCCAGGCATCACGGTGATAACTGTGAGTGTTTAAGAACAGGGGCAGTAGGGCTGGGTGTGGTGATTCACACCTGTATTCCTAGCACTTTGGGAGGCCAAGGCAGGTGGATCACCTGAGGTCAAGAGTTTGAGACTAGCCTGGCCAACATGGTGAAACCCCATCTCTACTAAAAATACAAAAATTAGCCCGGCATGATAGTGCGTGCCTATAATCCCAGCTACTTGGGAGGCTGAGGCAGGAGAATCCCTTGAACCTGGGAGGTGGGGGTTGCAGTGAGCCAAGATCGTGCCACTGCACTCCAGCCTGGGTGACAAGAGTGAAACTCCGTCTCAAAAACAAAAACAAAAACAAGGGCAGTATAGACTTTGGTTGGGTTCAGCCTTTGTTGGTTACTTGAGGCATGATACTTGAGCAACTTGTGCCTCAGTTTCTTTATCTGTAAAACAGGGAACAGTGCTCAAACTCATAGGATGGTTGGTAGCATTTCATGACATGATGCTAATAAAGTGGCTAGAACAGTGCCTGGCCTTCAGGAAGCATTCAATAAATATTAGTTTTATTGACATAGGAGAAATAGCTTAGAACAAAGTACCCTGGGTTAGTTCTAAGGAAGGCTGAGGGTTACCTGTTGTGGCATAGTCAATTGAAGTGAGTTACAATGAAGGCAGAGAAGGCAGAATTTGATAGAACTTGAAGAGTATGTATCAAGTGCTGCCCATGTGAACAGCTCTCTGCTGGAAGAAGGAGATTGAGGCTGAGCACAGGCTCACGCCTGTAATCCCAGCACTTTGGGAGGCCAAGGTGGGTGGATCACTTGAGGCCAGGAGTTTGAGACCAGCTGGGCCAACATGGTGAAAACTCATCTCTACTAAAAATACAAAAATTAGCTGGGCTTGGTGGTCCGCGCCTGTAATCCCAGCGTCTTGGGTGGCTGATGCACGAGAATTGCTTGAACCCCAGGAAGCGGAGGTTGCAGTGAGCCGAGATCGCACCACTGCACTCCAGCCTGGGGGACAGAGTGAGTAAAAGACCAAGCAGAGTCATTGAGTAAAAGACCTAGCAGAGTAAATCTAACGGATGAGTGTGTGGAAAGATGAGGTGCAGGAAAGCCAAACTCAAGATGGAGAAGGAGAAAAGAGAAACAGATGACCTACTCCTATTATGCCTTTTTTTGCTTGACATTTCCCTCCTGTCCCTGCTTTGCAGAGTAAAGACAGATGGTAATTTCCAGGTAGCCCAGGAGCCCGGTGCCATTTTTCCAAGACTGACCAGCTTTGAAAAGACTCTTGCCTGCCCAGTCTCCAGACATTCGCAAACTCTTCCCAAAGACAAACCCTTTCCAAAGCCCTGCTCTAAAACCGACCCAGCTCTTAGGAGGTGAGGCCAGGGCTCGCCACTGAAAGGGAGCAATTGTCCTCTCAACTGTCTGTGGGTCCAGCTGCCATAGGCCTGGAGGTGGTTCCACCACAGCCTGTGGTGCGGTCCGCCTGCAGGCTCTGATTCACTCCAGGGACAACACTTAAAAACGCAAGCAGGCAACACATGCGATTCCAGGAGCCCCTGCCAGATTCATACTCTGTTTTTCCATCATTTCACTGGCCTCCCCCTCAGGAGGGAGGCTCCTGAGGCATTTTACAAATAAAAGAATCAAACCCACCAACAGGACGAGCACCTGATAAGCAGTGAGGCATTGCCCAAGAGTTCCCAGGAAGAGGCGGCTTTGTTCTCCTTTCAAGCCCTTGACACTGCTAACAGCTCTAATCACTGCATTTGCCATGGGATGTAAAAGAAACCCACATTCCAGAAACCATGTAGTCTTCCCGGGAGAGACATGCAGGGAAAGGTGCCCAGGTCTGCAGGAACGTGTTCTTTCATTCCTGGTCCCTAACCTTGGCAGCTGTTCTGAGAACTTAACAGTGCCTCCTTTATGTACATGAAGGAGGGCACTTTTGCACATGTAGGGATGCATCTCACTACCCATATCCCCAAATCTGGTCTGCATATTTGACTACATTTCCTGCAAAACATACTTCTTAAGAACATGAAGATCTTCAAAATTCATCCAAGCCATTTTGGGGGAAATGAGGACTTCAGAAGTTTCTGAGGCTAGAAGCAAAACCGTATAGAACTTCTTCCCACATAATTTAACTATAAAATCTGATGATGTAACACCCACCCACACCCACATCACAAACCAAAGTGACATCACAAATACTAGAGATAATGGCTCAAAGCTGCCTGCATTGAGCCCACATCTAAACAAACAGCAATTTGCCCATATGTGACTAAGGCAGAGAAAGACAGGTGACTAGAAATATTAAAACAAAAACAAAAAAAAAGAGAGAGAGAGAAAAAGGAAAGAGAAAATACATATTTCTTCCCCTTACTAAGCTAGCGAAAATAAAATTTAAGGCAACTGAAATATGGTAAGGTGATTTTATTTCTATAGATGACATTTCTTAGAAAGCCAGATCTCTTTAAATATGTGATTCTAATCTATCCAAATTGTTCTGTTGCTCTTAGTTGGTCCCAGGAAATGGAGGCCGGCAGCATTAATGTTTTCGTGTTGCTTTTATGGTCCTCAGAGTATATTGATTTCCCTGTGAATGAGCTCTTCTTGGCGCATGGCAGTTGGTTAGAAAGTGTATTCAGCCGAACATGGTGGCTCATGCTGGTAATCCCAGCACTTTGGGAGGCTGAGGCGGGCGGATCATCTGAGGTGGGGAGTTCGAGGCCAGCCTGACCAACATGGAGAAACCCTGTCTCTACTAAAAATACAACATCAGCTGGGTGCGGTGGTGCATGCCTGTAATCTCAACTACTTCGGAGGCTGAGGCAGAAGAATCCCTTGAGCCCGGGAGGCGGAGGTTGCAGTGAGCCAAGATCACGCCATTGCACTCCAGCCTGGGCAACAAGAGCTAAACTCCGTCCCCCCCACACACAAAAAAAGAAAGTGTAAAGCGTATTCAGGGCTTGGCAGAACTTACCGGCTGTCATGTCTGTGTTGCAGCATGGTTGAGTGTCTACAGCTCATATGGAAGTCTTGAGTGACCCTGCCTTCCGTATCTTTCCAGCAGGCAGGAATTGTCTTCAATTCCTTGTGAACCCTCCCTGACCCCCGACTTCCTGCCCTGGCATGCTATGTTCTTTGAGAGCCTGAATATTACTGTAACCTTGTGGAGTGACTATCTCCTCCCTACTTGGTATAAAAGGCTTTGGTGAACAGGAGAGGACATATAGTAAATTTCTAAGAACTGGGTGGAAGGAATGGCCTATAAAAAGATGTTAATATTTTGCCTTCAAAGTAGACCTTTTATTCTCTGCTGAAGAACGGTCCTTTTAAAATTCATTTCTCCCTCCTTATAATTCTAGCACCCAATCAATGTGAGACGAGACCCCAGCATCCCTATCTATGGACTCCGACAGTCCATCTTATTAAATACCAGGCTTCAGGACTGCTATGTGGACTCGCCGGCTCTCACCAACATCTGGATGGCCAGAACATGTGCAAAGCAGAACATTAATGCCCCAGCACCAGCTACCACTTCCTCTTGGGAAGTTGTAAGGAACCCATTAATTGCCAGTTCCTTCTCCCTGGTTAAGCTTGTACTCAGGCGGCAACTGAAGAATAAGTGCTGCCCACCACCATGCAAGTTTGGAGAAGGAAAACTCTCGAAGAGATTAAAGCACAAGGACGATTCAGTGATGAAAGCCACCCAGCAGGCCAGGAAAAGAAACTTCATCAGTTCCAAGAGCAAGCAGCCAGCAGGGCATAGGAGGCCTGCAGGAGGCATCAGAGAGGTAGACGAGGACCAGTGGTGGGGAAAATGTGACTGGCGAGGGGAGGGAAAAGGTAAAGCCAGAGAACACTTAACGGTTGAGATTATATGGCGTGCGGCTACCGCCTATAATCTCAGCACTTTGGGAGGCCAAGCAGGGAGGATCACTTGAGCTCAGAGAGTTCAAGACCAGCCTGGGCAACATAGGGAGACCTCATCTCTACTAAAAAAAAAAAAAAAAAGCCAGGCATGATGCACATGCCTGTGATCCCAGCTACTTGGGAGGCTAAGGAGGGAGGATTACTTGAACCCAGGAGGTTGAGGCTGCAGTGAGCCATGATTGCACTACTGCATTCCTGATTGGGCAACAGAGCAAGACCCTGTCTAAAAAATAAAAAAAAAAAAAAAAAAAAAGAAAAGAAAAAAAGATCTATACCAAGTGAAGGAAATCCTAAAGTTGAAGACAGAATGTGGGGAGGAGGTGTAGCATGACAGAGTGGAAGAATGCCCCAGATTTTGGGGTCATGCTATTTATCATTTACACTATGCTTAGTTCTAGAGAAGGATGAGGTGGCTTTCATAAGCACAGACTTGACATGAAGGAAAAATAAGGGTGAGAAGGAAGGCTAGGATATGGGTAGTGCATGCTTGTGCACACACACACACACACACACAACCAGTGGTCAGCTATGGATCTGGATTGTGAGCGTCCACATGGAGAACTTGGCCTACGAAGAAGGCCTGATTGGAATCCAGGTATGTTACCCCGTGGTGGCCAGCAGGCCATCAGGGTTCAAATCCCAACTCGGACTGTAATTGAGGACGACCTCGGGCTTAATTTCTTTGTGCCTTGATATTATCATATTAAATGGGGGATAACGGCCCTTCCCTCATAGGGCTTCATGAATGTTAAATAATACATGCAAAAGGCTTTTGAACAGAGCCTGGCAGAGGGTAAATAATTAGGATCTTGCGTGGCATTCAATAGATGACAGCTGTTATTAACCTTCTAGCTGTGTGGCCTTGGGCAAACCAATTCCAATTTCTTTGCCTATAAAGTTGTCTCCTGCTGACCTCACAGGGTTATGGGGAACACATGCAGTGGTATGTACATAAAGGAATTCAGTAAACTAGAACATGCTAGACAAATGTTGGTTGTTGCTGTAACCAAGACTCTGGGACACAAATCCCAGATAATTATTCTGAGCTCCCCTCTCAAGTTTTAGCTTTGACTTCTCTGCTTTCTTCCTGGCTAGAGTAAAGAAAGTTCAAAGGAGAAAAAACTAACAGTCCGCCAAGATCTTGAGGACAGATATGCTGAACATGTGGCTGCCACCCAAGCGCTACCCCAGGACAGTGGGACAGCAGCCTGGAAGGGCCGAGTGTTGCTTCCTGAAACCCAAAAGAGACAGCAGTTGTCGGAGGACACGCTAACCATCCATGGTCTCCCCACAGAGGGTTACCAGGCTCTGTACCACGCTGTGGTGGAGCCAATGCTGTGGAATCCTTCAGGGACCCCCAAGAGGTACAGCCTGGAGCTGGGCAAGGCCATTAAACAAAAGCTCTGGGAGGCTCTTTGCAGTCAGGGTGCCATCTCTGAAGGTGCTCAGAGGGACCGGTTCCCTGGCAGGAAGCAGCCAGGTGTCCACGAGGAGCCTGTACTCAAGAAATGGCCCAAGTTAAAGAGCAAAAAATAGGAACAAGAACTCATGGGATTAGAGTATTCTCTGCTAGAGATCTGAAAAATAAACACAACTTTGCACCTTC\",\n          \"GATGTCGCTGGAGGTCTCGGAGAGGGGCGTATAGCCTCTTGTAGCGGGGGAGTCGTCCCGGTGCAAGGTGCGTCGATGAAGGACTTGGGGCGGGATCCCAGGAGGTACTGGGGAGCAGGGGTCGTCCCGGCCCGGGAGGGGTCGCTGGGGACTTGGGGACGGGATCCCAGGAGGTACCGGGGAGCAGGGGGCCTCCTGGCCCGTGGGGAGTCGCTGGGGATTTTGGGACGGGATCCTAGGAGGTACCGGGGAGCAGAGGACGTCCCGGCCCGGGGGAAGTCGCTGGGGACTTGGGACGGGATCCCAGGAGGTACCGGGGAGAGGGGGCGTCCCGGCCTGGTTGGGGTCGCTGAGGGACTCGGGCAGCGGGTCCCAGATCCCGAGACAGAGAAGATGTCCCAGCCCGGGGGGCGTCGCAGAGGGTCTTGGAAGACTCAAGGCGGAGAGGGCCTCCTGGGCTGAGGGGCGTCGCTGAGGAGCACCGGGTCCTGCAGCCGGGGGACAGTACCGGGCCCGGGGGCGTCGATGACGGGCCTGGGGGTAGCCCGGGTCCCAGGAAGGGACCATCCCGGCCTGAGGGGCGTCGCCTAGGGGCGAGGGGAAGGGGCCCGGAAGGTCCTGAAGCGGGAGGGAGGGGGCTCTCAGGCCGGCCGGCCCTTGTCGCCCCTCGAGTGGGCTCCGCTCAGGGCGGGCGGCGGGGGGTCGCTCACACTCACCGGGTCCGGGGCCGAGGCCGGGGCTAGGCGCGGGCCGGGCCGCCAGCGTGAGCGCCGCGGCGGAGAGCAGGGCGAGGCGGGCGGCTGGCGGCGCCATGGGCCGTCAGTGCGGGGCCCCGGGGCGCGGCCCGGGGTAAAGGGGGCAGCGCGGGGAGCCGGGGGCGGCCATGGGAGCAGGACGCGGGGCCCGAGTGTAGCCCCGCCCCGGCCGCCCGGCCGCCCCGCCCCGGCCCGCCCAGCTCAGCGCCGCCACCGCCCCTGCAGCCTGGGGCCCGCCCGGATCCGCGGAGAGGCCCGGGGGGCGGAGCCCACTGCGCACGCACGCCCCGCCCCGGCCGCGCGCTGCTCCTGCCACTCACGCATCAGGCCCCGCCCATCTCTCGCCTCACCTGGGCCGGCCCAGCAGGGCCCTCTGTTTGTCCCATCCAAGGGTAGGATCAGCAGGTGAGGGGCGGGGCCTCGGCCCGGCCTCCAAGTCACCAAGCGGTTTGGCCGGGCTGAAATAGGCCGGGCTCCCAGTGACCCCTGCTCCACCCTCACTTAGGTCGTGACCCACGCGCGAGCTCAGCTACAGCGGTAATAATTACAACAGCAGTTTGCAAACTGTTTTCCCTTGCTTAGCTCATCGTAATGGTTAAAAAAAGAATGCTTTGGAGAGTTCACATCCCACTTGGTCACATGTAACACCTTGGACAGGTTTAAAGGAAAAATAGCTCTTACCTCAATTTCCTCATCTGTTAAATGGAGGTGATAATAGCTGCTACTCCTCAGAGCGCTTGCACTAAATAAGAATGCCTACAGCCGAGTGCGGTGGCTAACAACTCCTGTGATCCCAGCACTTTGGGAGGCCGAGACGGAAGAATCGCTTGAGCCCAGGATTCTGAGACCAACCTGGGCAACACAGCGAGACCCCATCTCTACAAGAAAAAAATAAATAACTCAGTGGTCCCAGCTACTCAGGAGACTGAAGCAGGAGGATCGCTTGAGCCCTGGAGGTCGGGCTTGCAGTGAGCCTTCTTCGCACCACTGCACTCCAGCCTGGGTGACAGAGCAAGACCCTGTCTCAAAAACAAAAACAAAAACAAAAAAAAACCAAAACGAAAGCATGGAAGCTTTTACGCACGCAGAGTAGGAGTTCAGCAAATGTTGAATTTCTTTCTCCACCCCAAGTTTGCAGCCACATACTGACCCCTGGTTCCGTATCTTATCACATAGCCAGTTTCAACCAGGGCAACTTTCCTGATCTGTCTGAGGCTCAGCCGTTTGCAGAATAGGAATAATAAAAGGCCTTCAGAGGATTCGATGAGATAGAAGCAAGTAAAACGCCCAACGCAACGTCCAGAAATGGATGAGACAGTACTGCTTATCCTACAGCCTGGGATGTTAGGGATCAGAGAGCGCGCGGCTTGCTTAAACTTTCGGCCCCAGCGTGGTAGGCCCTTTGCCCGCCTCCCCCCGCCCCGACCCCGGCAAGGTCTTCCCCTGCCGCACCCGGGGTCGGTCGTGGGCCGTCCTCCTCCCCCGCCCTCCCGCCAGGCAGAGGGGATTTTCCTACAGTTCCCGCCTAAAGCGCAGCGGTGACGCCCGGCGGGGCAGGTCCAGGCGGCGCGCCGGGCGCTTTCAAGTCCCTGGTCCCGGGCTTGCAAGTGCAGACCTGCGCCCGGCGCGGCGGCGGGAACATCTGCACCAGTGTGCAGACTCGCCGCACCCAGAGCGCGGCTGGGCGGGACCGGCACGCCTTGGGTACACTCGCCCCTGGGCCCCCTCAACCCCCGGGCGCCACCCCCTTATCCACTCCCAGAGCCCGCGTACCCCCAAACTCCTTTCCCCACCCACGGCTGCTCGCAGGGCCGGGCCCCAAGGGCGCAGCTACCTCCCACGAGGCCTCTCGATCTCTGTCCCTCTCCACCCACACGACTCAGCGCTGTAGCAAGAAGAATCTTGTTCAGTTATTGTTTTCATGTGTCATTCCCGAGTTCAAGGATCCTCTGGGCCTCTTATTGGAACTTAGGGGCACCCACCATCCCCTCACCCAGCCAGCCTCGGTCCTGGGGACCGCCCCCCCCACCTCCTGCCACCGGTGATCCCCACTTTTGTCCAGCTTTCCCCTTTTCCACAATTTTGGGCCCTTCCCACCTTCTCCACCGGACCTGAGAACCCTTATTCCAGGACTTCTGTGCCCTCCCAGACAGCAGTGCCTTTGAAGAGAATGAAATGAGAAGATGCAGGTGAAGCTCTCAGCCCAAAGCAAACACTCAATAAAGTCGAACTTTCCTTCGTTCAGCAAATATTCAGCGAGGAACATGCCAAGGGCTGAGGATACAGCCACCAAGACCAAACCTTCCTCTTGGAACTTACATTCTAGTCAACCTGAAAACACACAAATTCCGAAGAACAAAAAGTACTGCAAAGACAAGAAAATTATGTTGACCACATGTTCTGTGGATTTTTTTTCTTTTTCTTTTTCTGGAGACAGGGCCTCCGTCTGTCGCCCACACTGGAGTGCAGTGGTGAGATCATGGCTCACTGCAGCCTCCAATTCCTGGGCTCCAGTGATCCTCCCGCTTCAGCCTCCTGGATAGCTGGGACTACAGAGGCATGCCACCACACCTGGCTAACCTTTGGACTACTTTAAGCAGGATGCCAGAGAAAGACTTCAGCAAAGCAGGACACAGGTCTCTCCAGTTCACAAGGTTCTTTCTGCATCTCATTCAGGTTTCAGAGGAGAAAATAGGAAGGGCACCCAGGAGGTTGGAAACAAAGTTTATTTGGCCAAGTCTGGGCAGCCATGCCTTGTGATAGTAGCTTTGGTCTGTCCTGCCTGCTGCGGGAAGGCACCCCTCCTGGGCTGGCAGCAGCCCCTCCTCCAATTGGCTCTCTCTGTCATGTACACTGCTTAGGACCAGCCATATAATTGGCATGACACAGTGAGAAATGAAAATGTATGTCCTTTGTCCAAACTTATTAAGAATTTCAAGACAGGCCAGGCGCTGTGGCTCACGCCTGTAATCCCAACATTTAGGGAGGCCAAGGCAGGTGGATCACCTGAGTTCAGGAGTTTGAGACCAGCCTGGCCAACATGGTGACACCTCATCTCTACTAAAAACACAAAAAATTATCCGGGCTTGGCGGTAGGTGCCTGTAATCCCAGCTACTCAGGAGGCTGAGACAGGAGAATTGCTTGAACCTGGGAGGTGGAGGTTGCAGTGAGCCAAGATCGTGCCACTGCACTCCAGCCTGAGTGACAGAGCAAGACTCCATCTCAAAAAAAAAAAAAAAAAAAAGAATTTCAAGACAGTGACAGCAGAAGATGAAAAGATAAAGCATGTGTGGGGCCCTTCCAAGCAAGGGGGCCTATGTTGATATACAGGTTGCACGCCCATGAAAATGGTCCTATTGTACAACAGGACTCAGAAATTTGTGTGCAAGCTCCTCCTGGCACTGGCTGCAGTGAAAAACATCCCTCTAACCTTCTCCACCCCCACCTTCTCTTTTCCAGCTGAAAGCCCAGCACAGGTGATGGAAACAGTGCACAAGAGGCTAAAAATACAGCCCAGCACGTCTAGGTCATGGCACTTAACAGCTCAAGGTCTTTTGTGGTCCTGGCTTCTGGCAGAGAGCAACCCGGGATACTCATTTCAGTGCTTTACATGCCTGGGGTGTTTAGCAGCTTTATTTCATGGAAAGAGTGAGGACACTGAATGCAGCTCCTGGGCCTCCAGAAAGATGTTGGTGTCCGTTTGTCTGTTTGCACTGTGCAAGTAAGGCCCCTTTCTTCCTCCGAGGGCAGCAGGATTGGCCAAGAAGCCAGGAGACCTACAATGTTGGAGTTTGGGTCATTCAACTGTCTTTGAAACTCCAGTTGGGATGATTTCTGCTCAAAGGGCAGTCATTTTATGGAAAGAGCCCTGGAGGAGGGGGCCTGATTATAATGACAGCAATAGCAAATGTTTACTGCAGACTTGACTAAGGCTATGCAGGGTGTCTACGGTTGCATTTCACAAAATCCGCCCAGTAGCCTTATGAGGGAGGCATTCTTACCAGCCCCACACCCACCCCAGAACAAAGAGAGATTAAATGACTTGCCCAAAGGTGACATCGTGTTAGAGCTGGGCTTTGAACCCAGGCAGTGGAACCCTAGTCTTACCCTCACAAGTACCAGGCTGGGCCAGGGGTGGTGGCTCACACCTGTAATCCCAGCACTTTGGGAGGCCGAGGCAAGAGGATCACAAGGTCAGGAGTTCAAGACCAGCCTGACCAACATAGTGAAACCCCATCTCTACAAAAATACAAAAACTAGCCAGGTGTAGTGGCGGGCGCCTGTAGTCTCAGGTACTTGGGAGACTGAGGCAGGAGAATGGCTTGAACCCAGGAGACAGAGGTTGCAGTGAGCCGAGATCGTGTCACTGCCCTCCAGCCTGGGAGACAGACCAAGACTCCATCTCAAAAAAAAAAAAAAAAAAAAAAAGTACCAGGTTGAGCCACCTCCTTGTCTTTTTTGTTTTAAAGACAGGGTCTTGCCGGGCACGGTGGCTTACGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGTGGGCAGATCATGAGGTCAGGAGATCGAGACTATCCTGGCTAACATGGTGAAACCCCGTCTCTACTAAAAATACAAAAAATTAGCCGGGCATGGTGGCAGGTGCCTGTAGTCCCAGCTACTCAGGAGGCTAAGGCAGGAGAACGGTGTGAACCCGAGAGGCAGAGGTTGCAGTGAGCCAAGATCACACTACTGCACTCCAGCCCAGGCAACAGAGCAAGACTCCGTCTCAAAAAAAAAAAAAAAAGAAATTTAAAGACAGGGTCTTGAGCCCGGGCAGGATGGCTCGTGCCTGTAATCTCAGCACTTTGGGAGGCTGAAGCGGGTGGATCACTTGAGGTTAGGAGTTCGAGACCAGCCTGGCCAACATGGTGAAACCCCATCTGTAAAAAATAAATAAATAAATAAAAATTAGCCAGGTGTCAGGGCATGCGCCTATAATGTCAGCTACTTGGGAGGCTGAGGCAGGAGAATCACTTGAACCCGGGAGACAGAAGTTGTAGTGAGCAGAGATAGTGCCATTGTCCTCCAACCCAGGCAAGAGTGAGACTCTGTCTCAAAAAAAAAAAAAAAAAGACGGGGTTTGATATGTTTGATATGTTGCCCAGATTGGATTGCAGTGACTATTCACAATCATGGCACACTACAGCCTCGAATTCCTGGGCTCCAGCAATCCTCTCACCTCAGACTCAGTAGCTGGGGCTACAGATGCTTGCTACTGTACCTGATGCCTCCTTATTCTGTCATTAGGCCAGGTATGGTGGCTCACATTTGTAATCCCAGTACCTCGGGAAGCTGAGGTGGGAGGATCTCTTGAACCCAGTATTTTGAGGTTACAGTGAACGAGGATACAGCCACTGCAACCCAGCCTGGGCAACAAGAAAGCCAGACCCTGTCTCTAAAATTAATAATAATAGCCTGGGCACGGTGGCTCATGGTTGTAATCTCAGTACCTTGGGAGTCCGATGCTGGCAGATCACCTGAGGTCAGGAGTTCGAGTCCAGCCTGGCCAATATGGCAAAACCCCAGTTCTGCTAAAAATACAAAAATTAGCCGGGCGTGGTGGCTCATGCCTGTAATCCCAGCTACTTGGAAGGCTAACGCAGGGTTCAGCTACTTGAACCCTGGTGGTGGAAGTTGCAGTGAGCCAAGGTCATGTCACTGCACTCTAGCCTGGGCGAGAAAGCGAGAGTCTGTGTCAAAAAAGAAAGAAAAGAGAAGAAAAAAGAAAATTAATAATAATAATTAATAAAAAATAAAATAATAGGTTTCTGTCACCAAATGCTTACTGAGTGGTCCAGAGCTGGGACTGGAATGGGGCAAGTGAAGTATGTGCCTCAGGCACAGAGTTTAAGAGGATTTTTAGAAAACACAATAAAAAATCCAGTAATCAAGATAAATATTATTTAATAGATATTTTATACATTTAATAAATATATTTAGTAAATAGTATTTAATGCAATATTTTAAAATAATTTTTTAACATCTAAATGAATGCAAAAAAATCCATTATGAACAGACTGTCCCAGCTTTAAATGAAGATAGGATCTGTATTCCCGATTTTCCTTCAGCCTCGGGTTTGAGGCTGCACTGTATCATTACTGAGTCTCCCAGGATTCTGGCTCAGACAACTAGTCATGGGAAATTCAAGAGAGGCAGTCATTTGCTTGTTCTGATGGGGGTGAAGGAAGAGAAAGACTCTGGTCTTGAACATAATGAGTTAGAGTTGTCTGTGGGGCATCCAGGTGGGCACCTCTGAAGACTCACAGTAGGGTTTGGGTTAGCATTGAGGAGGCGTTGGCTGGGCGCGGTGGCTCCCGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGCAGGTGGATCACCTGAGGTCAGGAGTCCAAGACCAGCCTGGCTGACATGGTGAAACCCTGTCTCTACTAAAAATACAAAAATTAGCTGGGCATGGTGGTGAGTGCTTGTAATCCCAGTTACTCAGTAGGCTGAGGCAGGAGAATTGCTTGAACCCAGGAGGCGGAGGTTGCAGTGAGCCAAGATCGTGCCACTGCACTCCACCCTTGGCAGCAGAGCGAGACTGCATCTCAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGAAAAGAAAAGAAAAGAAAGAAAGAAAGAAAGGAGGAAGTGTGGGACATCTGCGGGAGTCAGGTGAGAACTGGTGGGAAGGCACACAGGACACCATTATTTTGCATACTTGGCCCATGATTGGTGCATATTCAGCACATGTGGTCCAAATCATTGGTCCTCCGAGTTTGCATCACAGTTCAGGAATCTTTTCCAATTCTTTCCCATCCAAAGTGTTTCTCTTTTGCATCATGAACCTCAGATATGTTCCCACATGGGTATGGAAGTTTACTGCTTGGGTGTTAAAACAAAAACATCTGAGGAAATTAAAAGCAACTAAATTCCACTAGAATTGCATGGGTGGGGCAGGGCCATAAAGGTGGAAGACAAGAATTAATAGAAATTGACTCAATTTCTTCTTTATTAGCTGATCCACCAGAAGGAGAATAAGAGCTAAGAGAAGCTAAAAAGAAGAAAGGTCGTAGTAAAGGAAAGACAAAAAAAAAAGGAAGAAGGAAGATTATACGAGCTAAAAATACTGTGAGGAAAGTTTGGCTAAACATCAAAGAAGGACAGATGTGTTTGCAAATAGAAGCGGTGGTTTAGATATGTCAGGAAGCACAGCTCCAAATGATGGGTGGAAAATAAAGGTTAGTGAGACTGAGCTCAGCAGAATAAAAAAGCTTCATGGGAGAATCAAACTAAGGAAGAGGTCTTTCAGCAACATGGAAGCCCTACTGCTTCAACCCCGAGTTCCCCAGATCAAGTGCTGGCACCCATGATGGAAACTCTTGCCATGGTTTTAGTACCCTGGACCAAGTAGTCATTCCATCCTGACTTTAAAATTCTAAACAGCCTTTGATGGGACAATCTCTGCTAAAGACTAACCACTTCCTTATCTTATCTTCAGCTACCTGCTTCCCTTTCCGTTTAACAAAGCATAGAATATTCTGAACAACTCCCGTAACCAGGGGATGTAGCTCCTCTACATTCAACACTTCTGTGAGGTCAATCGATCTGACAAATATATAATGAAGATGGATCATGAAAGTCTGATACCCAAGTCAGATCAGAAACTTTCCCTACTTCCAAGAGGCTCATTGCTTGGCCTCATACACTAGATTTTTTTTTCTTTGTTCTTTTTCTTTTTTGTAGAGATGGGGTCTTGCTTTGTTGCCCAAGCTCATCTTGAACTCCTGGATTCAAGTGATCCTTCTGCCTCAGCCTCCCAAAGTGCTGGGATTATAGGTGCGAGCCACCACACCGGCCATAAACCAGATTGATGTACATGAATGACCTGAGACAAAGATCATATCAAATGTTGCATTACATCACTTAGACCATGAGCAACAGGAGCTCAAGAAAGGGTCAGATGAGTAAATGGACCATAATATTAATAACATTCACATGGTCAGGCGCAGTGGCTCACGCCTGAAATCCCAGCACTTTGGGAGGTGAAGGCTGGTAGATCACGAGGTCAGGAGTTCGAGACCAGCCTGGCCAACATGGTGAAACCCCCATCTCTGCTAAAAATACAAAAATTAGCCGGGCGTGGTGGTGCACACCTGTAATCTCAGCTACTCGGGAGGCTGAGGCAGGAGAATTGCTTGAACCCAGGAGGCGGAGGTTGCAGTGAGCTGAGAATGTGCCATTGCACTTCAGCCTGGGTGACAGATTGGGACTCCATCTCAAAAAAATAAATAAGTAAAAAATAACTTTCACATGGGTTAGGTGTGTTGGTTCACAGCTGTAGTCCCAGCACTTCGGGAGGCCAAGGAGGATCACTTGAGACCAGGAGTTCCAGACTAAGCTGAGCAACACAGGGAGACCTCATCTCTACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAATCACTGGGCAAAAACAAGCCTGTGATCCCAGCTACTTGGGAGGCCAAAGTGGGAGGATTGCTTGAGCCCAGGAGGTCGAGGCTCTCATGAGCCGTGATCACACCACTGCACTCCAGCCTGGGTGACAGACTGAGACCCTGTCTTGAAACAAAAACAAAAACACAAAAAAGCAAATATTCATTGAGTGCTTACTATATACACCAGGAGCCATGCCAAGTATTTCAGATGCATTGTCCCACTTCAAACTCACAATAACCCTTAGGTAGGCAGTATGACTAGGCCCATTTCACAGCTGAGGAAATTAAGGTTCAGGGGTATGAATTCACTTGCCTAATGAAAGATGGATCTGGGATGCCATCACAGGCAGATTCCCTCCTGGGCCTAGAGAAAGTCTCATGACAAGGACAGGACTAGAACTGGGCCCGGGTATGGTAGATGGATGGGCTGACCAACAGAGAGGCCAGGCATCACGGTGATAACTGTGAGTGTTTAAGAACAGGGGCAGTAGGGCTGGGTGTGGTGATTCACACCTGTATTCCTAGCACTTTGGGAGGCCAAGGCAGGTGGATCACCTGAGGTCAAGAGTTTGAGACTAGCCTGGCCAACATGGTGAAACCCCATCTCTACTAAAAATACAAAAATTAGCCCGGCATGATAGTGCGTGCCTATAATCCCAGCTACTTGGGAGGCTGAGGCAGGAGAATCCCTTGAACCTGGGAGGTGGGGGTTGCAGTGAGCCAAGATCGTGCCACTGCACTCCAGCCTGGGTGACAAGAGTGAAACTCCGTCTCAAAAACAAAAACAAAAACAAGGGCAGTATAGACTTTGGTTGGGTTCAGCCTTTGTTGGTTACTTGAGGCATGATACTTGAGCAACTTGTGCCTCAGTTTCTTTATCTGTAAAACAGGGAACAGTGCTCAAACTCATAGGATGGTTGGTAGCATTTCATGACATGATGCTAATAAAGTGGCTAGAACAGTGCCTGGCCTTCAGGAAGCATTCAATAAATATTAGTTTTATTGACATAGGAGAAATAGCTTAGAACAAAGTACCCTGGGTTAGTTCTAAGGAAGGCTGAGGGTTACCTGTTGTGGCATAGTCAATTGAAGTGAGTTACAATGAAGGCAGAGAAGGCAGAATTTGATAGAACTTGAAGAGTATGTATCAAGTGCTGCCCATGTGAACAGCTCTCTGCTGGAAGAAGGAGATTGAGGCTGAGCACAGGCTCACGCCTGTAATCCCAGCACTTTGGGAGGCCAAGGTGGGTGGATCACTTGAGGCCAGGAGTTTGAGACCAGCTGGGCCAACATGGTGAAAACTCATCTCTACTAAAAATACAAAAATTAGCTGGGCTTGGTGGTCCGCGCCTGTAATCCCAGCGTCTTGGGTGGCTGATGCACGAGAATTGCTTGAACCCCAGGAAGCGGAGGTTGCAGTGAGCCGAGATCGCACCACTGCACTCCAGCCTGGGGGACAGAGTGAGTAAAAGACCAAGCAGAGTCATTGAGTAAAAGACCTAGCAGAGTAAATCTAACGGATGAGTGTGTGGAAAGATGAGGTGCAGGAAAGCCAAACTCAAGATGGAGAAGGAGAAAAGAGAAACAGATGACCTACTCCTATTATGCCTTTTTTTGCTTGACATTTCCCTCCTGTCCCTGCTTTGCAGAGTAAAGACAGATGGTAATTTCCAGGTAGCCCAGGAGCCCGGTGCCATTTTTCCAAGACTGACCAGCTTTGAAAAGACTCTTGCCTGCCCAGTCTCCAGACATTCGCAAACTCTTCCCAAAGACAAACCCTTTCCAAAGCCCTGCTCTAAAATCGACCCAGCTCTTAGGAGGTGAGGCCAGGGCTCGCCACTGAAAGGGAGCAATTGTCCTCTCAACTGTCTGTGGGTCCAGCTGCCATAGGCCTGGAGGTGGTTCCACCACAGCCTGTGGTGCGGTCCGCCTGCAGGCTCTGATTCACTCCAGGGACAACACTTAAAAACGCAAGCAGGCAACACATGCGATTCCAGGAGCCCCTGCCAGATTCATACTCTGTTTTTCCATCATTTCACTGGCCTCCCCCTCAGGAGGGAGGCTCCTGAGGCATTTTACAAATAAAAGAATCAAACCCACCAACAGGACGAGCACCTGATAAGCAGTGAGGCATTGCCCAAGAGTTCCCAGGAAGAGGCGGCTTTGTTCTCCTTTCAAGCCCTTGACACTGCTAACAGCTCTAATTACTGCATTTGCCATGGGATGTAAAAGAAACCCACATTCCAGAAACCATGTAGTCTTCCCGGGAGAGACATGCAGGGAAAGGTGCCCAGGTCTGCAGGAACGTGTTCTTTCATTCCTGGTCCCTAACCTTGGCAGCTGTTCTGAGAACTTAACAGTGCCTCCTTTATGTACATGAAGGAGGGCACTTTTGCACATGTAGGGATGCATCTCACTACCCATATCCCCAAATCTGGTCTGCATATTTGACTACATTTCCTGCAAAACATACTTCTTAAGAACATGAAGATCTTCAAAATTCATCCAAGCCATTTTGGGGGAAATGAGGACTTCAGAAGTTTCTGAGGCTAGAAGCAAAACCGTATAGAACTTCTTCCCACATAATTTAACTATAAAATCTGATGATGTAACACCCACCCACACCCACATCACAAACCAAAGTGACATCACAAATACTAGAGATAATGGCTCAAAGCTGCCTGCATTGAGCCCACATCTAAACAAACAGCAATTTGCCCATATGTGACTAAGGCAGAGAAAGACAGGTGACTAGAAATATTAAAACAAAAACAAAAAAAAAGAGAGAGAGAGAAAAAGGAAAGAGAAAATACATATTTCTTCCCCTTACTAAGCTAGCGAAAATAAAATTTAAGGCAACTGAAATATGGTAAGGTGATTTTATTTCTATAGATGACATTTCTTAGAAAGCCAGATCTCTTTAAATATGTGATTCTAATCTATCCAAATTGTTCTGTTGCTCTTAGTTGGTCCCAGGAAATGGAGGCCGGCAGCATTAATGTTTTCGTGTTGCTTTTATGGTCCCCAGAGTATATTGATTTCCCTGTGAATGAGCTCTTCTTGGTGCATGGCAGTTGGTTAGAAAGTGTATTCAGCCGAACATGGTGGCTCATGCTGGTAATCCCAGCACTTTGGGAGGCTGAGGCGGGCGGATCATCTGAGGTGGGGAGTTCGAGGCCAGCCTGACCAACATGGAGAAACCCTGTCTCTACTAAAAATACAACATCAGCTGGGTGCGGTGGTGCATGCCTGTAATCTCAACTACTTCGGAGGCTGAGGCAGAAGAATCCCTTGAGCCCGGGAGGCGGAGGTTGCAGTGAGCCAAGATCACGCCATTGCACTCCAGCCTGGGCAACAAGAGCTAAACTCCGTCCCCCCCACACACAAAAAAAGAAAGTGTAAAGCGTATTCAGGGCTTGGCAGAACTTACCGGCTGTCATGTCTGTGTTGCAGCATGGTTGAGTGTCTACAGCTCATATGGAAGTCTTGAGTGACCCTGCCGTCCGTATCTTTCCAGCAGGCAGGAATTGTCTTCAATTCCTTGTGAACCCTCCCTGACCCCCGACTTCCTGCCCTGGCATGCTATGTTCTTTGAGAGCCTGAATATTACTGTAACCTTGTGGAGTGACTATCTCCTCCCTACTTGGTATAAAAGGCTTTGGTGAACAGGAGAGGACATATAGTAAATTTCTAAGAACTGGGTGGAAGGAATGGCCTATAAAAAGATGTTAATATTTTGCCTTCAAAGTAGACCTTTTATTCTCTGCTGAAGAACGGTCCTTTTAAAATTCATTTCTCCCTCCTTATAATTCTAGCACCCAATCAATGTGAGACGAGACCCCAGCATCCCTATCTATGGACTCCGACAGTCCATCTTATTAAATACCAGGCTTCAGGACTGCTATGTGGACTCACCGGCTCTCACCAACATCTGGATGGCCAGAACATGTGCAAAGCAGAACATTAATGCCCCAGCACCAGCTACCACTTCCTCTTGGGAAGTTGTAAGGAACCCATTAATTGCCAGTTCCTTCTCCCTGGTTAAGCTTGTACTCAGGCGGCAACTGAAGAATAAGTGCTGCCCACCACCATGCAAGTTTGGAGAAGGAAAACTCTCGAAGAGATTAAAGCACAAGGACGATTCAGTGATGAAAGCCACCCAGCAGGCCAGGAAAAGAAACTTCATCAGTTCCAAGAGCAAGCAGCCAGCAGGGCATAGGAGGCCTGCAGGAGGCATCAGAGAGGTAGACGAGGACCAGTGGTGGGGAAAATGTGACTGGCGAGGGGAGGGAAAAGGTAAAGCCAGAGAACACTTAACGGTTGAGATTATATGGCGTGCGGCTACCGCCTATAATCTCAGCACTTTGGGAGGCCAAGCAGGGAGGATCACTTGAGCTCAGAGAGTTCAAGACCAGCCTGGGCAACATAGGGAGACCTCATCTCTACTAAAAAAAAAAAAAAAAAGCCAGGCATGATGCACATGCCTGTGATCCCAGCTACTTGGGAGGCTAAGGAGGGAGGATTACTTGAACCCAGGAGGTTGAGGCTGCAGTGAGCCATGATTGCACTACTGCATTCCTGATTGGGCAACAGAGCAAGACCCTGTCTAAAAAATAAAAAAAAAAAAAAAAAAAAAAGAAAAGAAAAAAAGATCTATACCAAGTGAAGGAAATCCTAAAGTTGAAGACAGAATGTGGGGAGGAGGTGTAGCATGACAGAGTGGAAGAATGCCCCAGATTTTGGGGTCATGCTATTTATCATTTACACTATGCTTAGTTCTAGAGAAGGATGAGGTGGCTTTCATAAGCACAGACTTGACATGAAGGAAAAATAAGGGTGAGAAGGAAGGCTAGGATATGGGTAGTGCATGCTTGTGCACACACACACACACACACACAACCAGTGGTCAGCTATGGATCTGGATTGTGAGCGTCCACATGGAGAACTTGGCCTACGAAGGCCTGATTGGAATCCAGGTATGTTACCCCGTGGTGGCCAGCAGGCCATCAGGGTTCAAATCCCAACTCGGACTGTAATTGAGGACGACCTCGGGCTTAATTTCTTTGTGCCTTGATATTATCATATTAAATGGGGGATAACGGCCCTTCCCTCATAGGGCTTCATGAATGTTAAATAATACATGCAAAAGGCTTTTGAACAGAGCCTGGCAGAGGGTAAATAATTAGGATCTTGCGTGGCATTCAATAGATGACAGCTGTTATTAACCTTCTAGCTGTGTGGCCTTGGGCAAACCAATTCCAATTTCTTTGCCTATAAAGTTGTCTCCTGCTGACCTCACAGGGTTATGGGGAACACATGCAGTGGTATGTACATAAAGGAATTCAGTAAACTAGAACATGCTAGACAAATGTTGGTTGTTGCTGTAACCAAGACTCTGGGACACAAATCCCAGATAATTATTCTGAGCTCCCCTCTCAAGTTTTAGCTTTGACTTCTCTGCTTTCTTCCTGGCTAGAGTAAAGAAAGTTCAAAGGAGAAAAAACTAACAGTCCGCCAAGATCTTGAGGACAGATATGCTGAACATGTGGCTGCCACCCAAGCGCTACCCCAGGACAGTGGGACAGCAGCCTGGAAGGGCCGAGTGTTGCTTCCTGAAACCCAAAAGAGACAGCAGTTGTCGGAGGACACGCTAACCATCCATGGTCTCCCCACAGAGGGTTACCAGGCTCTGTACCACGCTGTGGTGGAGCCAATGCTGTGGAATCCTTCAGGGACCCCCAAGAGGTACAGCCTGGAGCTGGGCAAGGCCATTAAACAAAAGCTCTGGGAGGCTCTTTGCAGTCAGGGTGCCATCTCTGAAGGTGCTCAGAGGGACCGGTTCCCTGGCAGGAAGCAGCCAGGTGTCCACGAGGAGCCTGTACTCAAGAAATGGCCCAAGTTAAAGAGCAAAAAATAGGAACAAGAACTCATGGGATTAGAGTATTCTCTGCTAGAGATCTGAAAAATAAACACAACTTTGCACCTTC\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "##### EDA중 c22orf31이 22번 염색체의 전체가 아닌것으로 확인\n",
        "##### 교재를 빠르게 확인 후 Deepchem 튜토리얼에서 나온 내용을 배우기로 학습방향 전환\n",
        "\n"
      ],
      "metadata": {
        "id": "ClXVvfAJjHmC"
      }
    }
  ]
}